<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sarah Tanja">
<meta name="dcterms.date" content="2025-10-17">

<title>Compositional Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="compositional_analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="compositional_analysis_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="compositional_analysis_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="compositional_analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="compositional_analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="compositional_analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="compositional_analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="compositional_analysis_files/libs/quarto-html/quarto-syntax-highlighting-4a990d8dcb58f517c7c86712b8f2ac7c.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="compositional_analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="compositional_analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="compositional_analysis_files/libs/bootstrap/bootstrap-9af6de2548c106c4a05dbc3b1b139c6f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>
<link href="compositional_analysis_files/libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">

<script src="compositional_analysis_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>

<script src="compositional_analysis_files/libs/plotly-binding-4.11.0/plotly.js"></script>

<script src="compositional_analysis_files/libs/typedarray-0.1/typedarray.min.js"></script>

<script src="compositional_analysis_files/libs/jquery-3.5.1/jquery.min.js"></script>

<link href="compositional_analysis_files/libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet">

<script src="compositional_analysis_files/libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>

<link href="compositional_analysis_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">

<script src="compositional_analysis_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>



</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background"><span class="header-section-number">1</span> Background</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup"><span class="header-section-number">2</span> Setup</a>
  <ul class="collapse">
  <li><a href="#install-load-packages" id="toc-install-load-packages" class="nav-link" data-scroll-target="#install-load-packages"><span class="header-section-number">2.1</span> Install &amp; load packages</a></li>
  </ul></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data"><span class="header-section-number">3</span> Load Data</a>
  <ul class="collapse">
  <li><a href="#load-metadata" id="toc-load-metadata" class="nav-link" data-scroll-target="#load-metadata"><span class="header-section-number">3.1</span> Load metadata</a></li>
  <li><a href="#import-qiime2-artifacts" id="toc-import-qiime2-artifacts" class="nav-link" data-scroll-target="#import-qiime2-artifacts"><span class="header-section-number">3.2</span> Import QIIME2 artifacts</a></li>
  <li><a href="#prepare-data-for-analysis" id="toc-prepare-data-for-analysis" class="nav-link" data-scroll-target="#prepare-data-for-analysis"><span class="header-section-number">3.3</span> Prepare data for analysis</a></li>
  </ul></li>
  <li><a href="#clr-transformation" id="toc-clr-transformation" class="nav-link" data-scroll-target="#clr-transformation"><span class="header-section-number">4</span> CLR Transformation</a></li>
  <li><a href="#aitchison-distance" id="toc-aitchison-distance" class="nav-link" data-scroll-target="#aitchison-distance"><span class="header-section-number">5</span> Aitchison Distance</a></li>
  <li><a href="#pca-biplots" id="toc-pca-biplots" class="nav-link" data-scroll-target="#pca-biplots"><span class="header-section-number">6</span> PCA Biplots</a>
  <ul class="collapse">
  <li><a href="#plot-stage-x-leachate-x-cross" id="toc-plot-stage-x-leachate-x-cross" class="nav-link" data-scroll-target="#plot-stage-x-leachate-x-cross"><span class="header-section-number">6.1</span> Plot stage x leachate x cross</a></li>
  <li><a href="#plot-stage-x-leachate-x-spawn-night" id="toc-plot-stage-x-leachate-x-spawn-night" class="nav-link" data-scroll-target="#plot-stage-x-leachate-x-spawn-night"><span class="header-section-number">6.2</span> Plot stage x leachate x spawn night</a></li>
  <li><a href="#interactive-plotly" id="toc-interactive-plotly" class="nav-link" data-scroll-target="#interactive-plotly"><span class="header-section-number">6.3</span> Interactive plotly</a></li>
  </ul></li>
  <li><a href="#permanova" id="toc-permanova" class="nav-link" data-scroll-target="#permanova"><span class="header-section-number">7</span> PERMANOVA</a>
  <ul class="collapse">
  <li><a href="#fit-betadispr" id="toc-fit-betadispr" class="nav-link" data-scroll-target="#fit-betadispr"><span class="header-section-number">7.1</span> Fit betadispr</a></li>
  <li><a href="#perform-permanova" id="toc-perform-permanova" class="nav-link" data-scroll-target="#perform-permanova"><span class="header-section-number">7.2</span> Perform PERMANOVA</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">8</span> Summary</a>
  <ul class="collapse">
  <li><a href="#key-findings" id="toc-key-findings" class="nav-link" data-scroll-target="#key-findings"><span class="header-section-number">8.1</span> Key Findings</a></li>
  <li><a href="#next-steps" id="toc-next-steps" class="nav-link" data-scroll-target="#next-steps"><span class="header-section-number">8.2</span> Next Steps</a></li>
  <li><a href="#output-files" id="toc-output-files" class="nav-link" data-scroll-target="#output-files"><span class="header-section-number">8.3</span> Output Files</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">9</span> Session Info</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="compositional_analysis.md"><i class="bi bi-file-code"></i>Github (GFM)</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Compositional Data Analysis</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Following Gloor et al.&nbsp;2017 recommendations for microbiome compositional data analysis</p>
</div>



<div class="quarto-title-meta column-page-right">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Sarah Tanja </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 17, 2025</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">October 22, 2025</p>
    </div>
  </div>
    
  </div>
  


</header>


<section id="background" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Background</h1>
<p>This analysis follows the recommendations from Gloor et al.&nbsp;2017 for compositional data analysis of microbiome data. Microbiome sequence count data are compositional - they represent relative abundances constrained to sum to a total (sequencing depth). This requires specialized statistical methods that account for the compositional nature of the data.</p>
<p>Key recommendations from Gloor et al.&nbsp;2017: - Use centered log-ratio (CLR) transformation to convert compositional data to Euclidean space - Calculate Aitchison distance (Euclidean distance on CLR-transformed data) for dissimilarity - Use variance-based methods like PCA on CLR-transformed data - Perform statistical tests on transformed data</p>
</section>
<section id="setup" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Setup</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file paths</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>feature_table_path <span class="ot">&lt;-</span> <span class="st">"../salipante/241121_StonyCoral/270x200/250414_StonyCoral_270x200_featureTable_filtered.qza"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>metadata_path <span class="ot">&lt;-</span> <span class="st">"../metadata/meta.csv"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>taxonomy_path <span class="ot">&lt;-</span> <span class="st">"../salipante/241121_StonyCoral/270x200/250414_270x200_representative-sequences_taxonomy.qza"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="install-load-packages" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="install-load-packages"><span class="header-section-number">2.1</span> Install &amp; load packages</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check R version</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># R.version.string</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Install and load required packages</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("BiocManager", quietly = TRUE)) {</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  install.packages("BiocManager")</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Install Bioconductor packages</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("phyloseq", quietly = TRUE)) {</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  BiocManager::install("phyloseq")</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("ANCOMBC", quietly = TRUE)) {</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  BiocManager::install("ANCOMBC")</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Install CRAN packages</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co">#required_packages &lt;- c("tidyverse", "vegan", "compositions", "zCompositions", </span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                       "ggplot2", "reshape2", "corrplot", "devtools")</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="co">#for (pkg in required_packages) {</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co">#  if (!requireNamespace(pkg, quietly = TRUE)) {</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co">#    install.packages(pkg)</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  }</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Install qiime2R from GitHub</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("qiime2R", quietly = TRUE)) {</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="co">#  devtools::install_github("jbisanz/qiime2R")</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Install SpiecEasi for SparCC (optional, can be slow to install)</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("SpiecEasi", quietly = TRUE)) {</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a><span class="co">#  devtools::install_github("zdk123/SpiecEasi")</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zCompositions)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Load Data</h1>
<section id="load-metadata" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="load-metadata"><span class="header-section-number">3.1</span> Load metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load metadata</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(metadata_path)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># set factors</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">collection_date =</span> <span class="fu">as.Date</span>(collection_date, <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">stage    =</span> <span class="fu">factor</span>(stage,    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"cleavage"</span>, <span class="st">"prawnchip"</span>, <span class="st">"earlygastrula"</span>), <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">leachate =</span> <span class="fu">factor</span>(leachate, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"control"</span>, <span class="st">"low"</span>, <span class="st">"mid"</span>, <span class="st">"high"</span>),        <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">spawn_night =</span> <span class="fu">factor</span>(</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>      collection_date,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels  =</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"06-Jul-2024"</span>, <span class="st">"07-Jul-2024"</span>, <span class="st">"08-Jul-2024"</span>), <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels  =</span> <span class="fu">c</span>(<span class="st">"July 6th"</span>, <span class="st">"July 7th"</span>, <span class="st">"July 8th"</span>),</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">ordered =</span> <span class="cn">TRUE</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co"># View metadata structure</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [63 × 9] (S3: tbl_df/tbl/data.frame)
 $ sample_id      : chr [1:63] "101112C14" "101112C4" "101112C9" "101112H14" ...
 $ collection_date: Date[1:63], format: "2024-07-08" "2024-07-08" ...
 $ parents        : num [1:63] 101112 101112 101112 101112 101112 ...
 $ group          : chr [1:63] "C14" "C4" "C9" "H14" ...
 $ hpf            : num [1:63] 14 4 9 14 4 9 14 4 9 14 ...
 $ stage          : Ord.factor w/ 3 levels "cleavage"&lt;"prawnchip"&lt;..: 3 1 2 3 1 2 3 1 2 3 ...
 $ leachate       : Ord.factor w/ 4 levels "control"&lt;"low"&lt;..: 1 1 1 4 4 4 2 2 2 3 ...
 $ leachate_mgL   : num [1:63] 0 0 0 1 1 1 0.01 0.01 0.01 0.1 ...
 $ spawn_night    : Ord.factor w/ 3 levels "July 6th"&lt;"July 7th"&lt;..: 3 3 3 3 3 3 3 3 3 3 ...</code></pre>
</div>
</div>
</section>
<section id="import-qiime2-artifacts" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="import-qiime2-artifacts"><span class="header-section-number">3.2</span> Import QIIME2 artifacts</h2>
<section id="feature-table" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="feature-table"><span class="header-section-number">3.2.1</span> Feature table</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load feature table from QIIME2 artifact</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>feature_table <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(feature_table_path)<span class="sc">$</span>data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxonomy" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="taxonomy"><span class="header-section-number">3.2.2</span> Taxonomy</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load taxonomy (optional, for annotation)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>taxonomy <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(taxonomy_path)<span class="sc">$</span>data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display data dimensions</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Feature table dimensions:"</span>, <span class="fu">dim</span>(feature_table), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Feature table dimensions: 6147 63 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of samples in metadata:"</span>, <span class="fu">nrow</span>(meta), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of samples in metadata: 63 </code></pre>
</div>
</div>
</section>
</section>
<section id="prepare-data-for-analysis" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="prepare-data-for-analysis"><span class="header-section-number">3.3</span> Prepare data for analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert feature table to matrix (samples as rows, features as columns)</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># QIIME2 format typically has features as rows, samples as columns</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>feature_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(feature_table)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>feature_matrix_t <span class="ot">&lt;-</span> <span class="fu">t</span>(feature_matrix)  <span class="co"># Transpose so samples are rows</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure sample IDs match between feature table and metadata</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>sample_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(feature_matrix_t)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>meta_filt <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">%in%</span> sample_ids) <span class="sc">%&gt;%</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(sample_id, sample_ids))</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Samples in feature table:"</span>, <span class="fu">nrow</span>(feature_matrix_t), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Samples in feature table: 63 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Samples in metadata:"</span>, <span class="fu">nrow</span>(meta_filt), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Samples in metadata: 63 </code></pre>
</div>
</div>
</section>
</section>
<section id="clr-transformation" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> CLR Transformation</h1>
<p>The centered log-ratio (CLR) transformation converts compositional data to Euclidean space. It handles zeros by replacing them with small values (geometric Bayesian multiplicative replacement).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace zeros with small values using cmultRepl from zCompositions</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This is necessary because log transformation is undefined for zeros</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>feature_matrix_nozero <span class="ot">&lt;-</span> <span class="fu">cmultRepl</span>(feature_matrix_t, <span class="at">method =</span> <span class="st">"CZM"</span>, <span class="at">label =</span> <span class="dv">0</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply CLR transformation</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>feature_clr <span class="ot">&lt;-</span> <span class="fu">clr</span>(feature_matrix_nozero)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame for easier handling</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>feature_clr_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(feature_clr)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"CLR-transformed data dimensions:"</span>, <span class="fu">dim</span>(feature_clr_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>CLR-transformed data dimensions: 63 663 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_clr_df[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 5%">
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 18%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">a04fbb7832640df06d3e5bcefe6e5801</th>
<th style="text-align: right;">10f0b9b94a3c8595e648259668664749</th>
<th style="text-align: right;">9a5dabc71f65a222f30b7415aa0d9dd0</th>
<th style="text-align: right;">cfb17500dcecfc7d15edf48a0c944272</th>
<th style="text-align: right;">069cf7a7c4b413f53230b4b140ff08c7</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">101112C14</td>
<td style="text-align: right;">0.2486652</td>
<td style="text-align: right;">4.709150</td>
<td style="text-align: right;">4.663244</td>
<td style="text-align: right;">1.1531215</td>
<td style="text-align: right;">2.912620</td>
</tr>
<tr class="even">
<td style="text-align: left;">101112C4</td>
<td style="text-align: right;">-3.7257422</td>
<td style="text-align: right;">5.497918</td>
<td style="text-align: right;">5.248164</td>
<td style="text-align: right;">-3.7257422</td>
<td style="text-align: right;">2.151650</td>
</tr>
<tr class="odd">
<td style="text-align: left;">101112C9</td>
<td style="text-align: right;">-0.3735785</td>
<td style="text-align: right;">3.776336</td>
<td style="text-align: right;">4.105380</td>
<td style="text-align: right;">-3.4432900</td>
<td style="text-align: right;">1.885204</td>
</tr>
<tr class="even">
<td style="text-align: left;">101112H14</td>
<td style="text-align: right;">-0.6744121</td>
<td style="text-align: right;">3.660698</td>
<td style="text-align: right;">4.517618</td>
<td style="text-align: right;">0.9678157</td>
<td style="text-align: right;">2.087705</td>
</tr>
<tr class="odd">
<td style="text-align: left;">101112H4</td>
<td style="text-align: right;">-3.9079152</td>
<td style="text-align: right;">5.443821</td>
<td style="text-align: right;">5.515943</td>
<td style="text-align: right;">0.3069533</td>
<td style="text-align: right;">1.925441</td>
</tr>
<tr class="even">
<td style="text-align: left;">101112H9</td>
<td style="text-align: right;">-3.2733059</td>
<td style="text-align: right;">4.203059</td>
<td style="text-align: right;">5.086768</td>
<td style="text-align: right;">-3.2733059</td>
<td style="text-align: right;">2.621114</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="aitchison-distance" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Aitchison Distance</h1>
<p>The Aitchison distance is the Euclidean distance between samples after CLR transformation. This is the appropriate distance metric for compositional data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Aitchison distance (Euclidean distance on CLR-transformed data)</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>aitchison_dist <span class="ot">&lt;-</span> <span class="fu">dist</span>(feature_clr, <span class="at">method =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Display distance matrix summary</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Aitchison distance matrix summary:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Aitchison distance matrix summary:</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">as.vector</span>(aitchison_dist))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  42.03   66.38   77.38   78.23   89.85  115.41 </code></pre>
</div>
</div>
</section>
<section id="pca-biplots" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> PCA Biplots</h1>
<p>Variance-based compositional PCA on CLR-transformed data. ## Perform PCA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on CLR-transformed data</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>pca_result <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(feature_clr, <span class="at">scale. =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate variance explained</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>variance_explained <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_result)<span class="sc">$</span>importance[<span class="dv">2</span>, ] <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pca_result<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pca_result<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample_id =</span> <span class="fu">rownames</span>(feature_clr)</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with metadata</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> pca_df <span class="sc">%&gt;%</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta_filt, <span class="at">by =</span> <span class="st">"sample_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plot-stage-x-leachate-x-cross" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="plot-stage-x-leachate-x-cross"><span class="header-section-number">6.1</span> Plot stage x leachate x cross</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>pca_plot_cross <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                   <span class="co"># inside color</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(parents),       <span class="co"># outline color</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate                <span class="co"># shape must be 21–25</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span>   <span class="co"># stroke controls outline width</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># light grey -&gt; dark grey for discrete stages (shows in legend)</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Parental cross"</span>,</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Parental cross"</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3) make the fill legend draw a filled shape with an outline</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>), </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>pca_plot_cross</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/compositional_pca_cross.png"</span>, pca_plot_cross, </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">7</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-stage-x-leachate-x-spawn-night" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="plot-stage-x-leachate-x-spawn-night"><span class="header-section-number">6.2</span> Plot stage x leachate x spawn night</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>pca_plot_spawn_night <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                   <span class="co"># inside color</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(spawn_night),       <span class="co"># outline color</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate                <span class="co"># shape must be 21–25</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span>   <span class="co"># stroke controls outline width</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># light grey -&gt; dark grey for discrete stages (shows in legend)</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Spawn night"</span>,</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Spawn night"</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3) make the fill legend draw a filled shape with an outline</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>), </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>pca_plot_spawn_night</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/compositional_pca_spawn_night.png"</span>, pca_plot_spawn_night, </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">7</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interactive-plotly" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="interactive-plotly"><span class="header-section-number">6.3</span> Interactive plotly</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># add a tooltip aesthetic to your existing ggplot</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                    <span class="co"># inside color</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(parents),        <span class="co"># outline color</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate,                <span class="co"># shape must be 21–25</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">paste0</span>(</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Stage: "</span>, stage,</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;br&gt;Leachate: "</span>, leachate,</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;br&gt;Parent: "</span>, parents</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Parental cross"</span>,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Parental cross"</span></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to plotly and use only the custom text in the tooltip</span></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf_interactive <span class="ot">&lt;-</span> <span class="fu">ggplotly</span>(pca_plot_hpf, <span class="at">tooltip =</span> <span class="st">"text"</span>)</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf_interactive</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-2bce70ef37abbb8009d7" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-2bce70ef37abbb8009d7">{"x":{"data":[{"x":[47.903501760984661],"y":[26.173885088862704],"text":"Stage: cleavage<br>Leachate: control<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(control,cleavage,123)","legendgroup":"(control,cleavage,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[30.778686066616888],"y":[21.057441778281888],"text":"Stage: cleavage<br>Leachate: control<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(control,cleavage,456)","legendgroup":"(control,cleavage,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-9.0716835458250813],"y":[12.634689548172277],"text":"Stage: cleavage<br>Leachate: control<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(control,cleavage,789)","legendgroup":"(control,cleavage,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-17.62840777536832],"y":[18.2391470245352],"text":"Stage: cleavage<br>Leachate: control<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(control,cleavage,101112)","legendgroup":"(control,cleavage,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-26.23571584157088],"y":[25.398215050304888],"text":"Stage: cleavage<br>Leachate: control<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(control,cleavage,131415)","legendgroup":"(control,cleavage,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[51.87112899126334],"y":[-7.4943755523967184],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 45","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(100,178,0,1)"}},"hoveron":"points","name":"(control,prawnchip,45)","legendgroup":"(control,prawnchip,45)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-14.795237540103564],"y":[5.5849772191514075],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 67","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,189,92,1)"}},"hoveron":"points","name":"(control,prawnchip,67)","legendgroup":"(control,prawnchip,67)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-8.7821816773731793],"y":[0.12945381754309732],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 89","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,193,167,1)"}},"hoveron":"points","name":"(control,prawnchip,89)","legendgroup":"(control,prawnchip,89)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[42.70835359672337],"y":[-12.534598446748662],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(control,prawnchip,123)","legendgroup":"(control,prawnchip,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-35.541453611077877],"y":[4.9574971060711936],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(control,prawnchip,101112)","legendgroup":"(control,prawnchip,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-37.292257490312501],"y":[7.3192425448471337],"text":"Stage: prawnchip<br>Leachate: control<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(control,prawnchip,131415)","legendgroup":"(control,prawnchip,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-26.252608727946651],"y":[-11.836973000624857],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 6","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(control,earlygastrula,6)","legendgroup":"(control,earlygastrula,6)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-20.123904887821684],"y":[-14.058490056040744],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 7","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(219,142,0,1)"}},"hoveron":"points","name":"(control,earlygastrula,7)","legendgroup":"(control,earlygastrula,7)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[39.482988711094826],"y":[-24.305339849818957],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 45","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(100,178,0,1)"}},"hoveron":"points","name":"(control,earlygastrula,45)","legendgroup":"(control,earlygastrula,45)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-21.748188325331402],"y":[-16.860392154256356],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 89","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,193,167,1)"}},"hoveron":"points","name":"(control,earlygastrula,89)","legendgroup":"(control,earlygastrula,89)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[29.848511167457509],"y":[-29.255197656247212],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(control,earlygastrula,123)","legendgroup":"(control,earlygastrula,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-33.713179780369032],"y":[-17.697425636709291],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(control,earlygastrula,101112)","legendgroup":"(control,earlygastrula,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-38.180874888186793],"y":[-12.345612818087835],"text":"Stage: earlygastrula<br>Leachate: control<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"circle","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(control,earlygastrula,131415)","legendgroup":"(control,earlygastrula,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[51.394871472463791],"y":[23.601764441768182],"text":"Stage: cleavage<br>Leachate: low<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(low,cleavage,123)","legendgroup":"(low,cleavage,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[29.580146035397728],"y":[23.220096137820867],"text":"Stage: cleavage<br>Leachate: low<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(low,cleavage,456)","legendgroup":"(low,cleavage,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-3.6987423121916336],"y":[18.469983383831661],"text":"Stage: cleavage<br>Leachate: low<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(low,cleavage,789)","legendgroup":"(low,cleavage,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-21.321708802639066],"y":[21.865709555900835],"text":"Stage: cleavage<br>Leachate: low<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(low,cleavage,101112)","legendgroup":"(low,cleavage,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-21.19830266338969],"y":[23.725436111692083],"text":"Stage: cleavage<br>Leachate: low<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(low,cleavage,131415)","legendgroup":"(low,cleavage,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[49.28183142559844],"y":[-3.8171862301404853],"text":"Stage: prawnchip<br>Leachate: low<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(low,prawnchip,123)","legendgroup":"(low,prawnchip,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[33.52230484075838],"y":[-3.7931799031176636],"text":"Stage: prawnchip<br>Leachate: low<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(low,prawnchip,456)","legendgroup":"(low,prawnchip,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-17.818076583419202],"y":[-4.0274255755013124],"text":"Stage: prawnchip<br>Leachate: low<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(low,prawnchip,789)","legendgroup":"(low,prawnchip,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-33.251772855570692],"y":[-3.4853824325527829],"text":"Stage: prawnchip<br>Leachate: low<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(low,prawnchip,101112)","legendgroup":"(low,prawnchip,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-39.933795227253817],"y":[5.2839624123654634],"text":"Stage: prawnchip<br>Leachate: low<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(low,prawnchip,131415)","legendgroup":"(low,prawnchip,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[41.022100957860154],"y":[-25.703961244779673],"text":"Stage: earlygastrula<br>Leachate: low<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(low,earlygastrula,123)","legendgroup":"(low,earlygastrula,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[20.578606996055143],"y":[-30.993666100896263],"text":"Stage: earlygastrula<br>Leachate: low<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(low,earlygastrula,456)","legendgroup":"(low,earlygastrula,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-13.968353531563169],"y":[-30.563134756142386],"text":"Stage: earlygastrula<br>Leachate: low<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(low,earlygastrula,789)","legendgroup":"(low,earlygastrula,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-37.820913899085674],"y":[-11.124846782773957],"text":"Stage: earlygastrula<br>Leachate: low<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(low,earlygastrula,101112)","legendgroup":"(low,earlygastrula,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-39.976303376669435],"y":[-16.292698316951284],"text":"Stage: earlygastrula<br>Leachate: low<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"square","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(low,earlygastrula,131415)","legendgroup":"(low,earlygastrula,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[46.485450586134846],"y":[22.902858513465425],"text":"Stage: cleavage<br>Leachate: mid<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(mid,cleavage,123)","legendgroup":"(mid,cleavage,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[39.592725915061813],"y":[21.024192280323209],"text":"Stage: cleavage<br>Leachate: mid<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(mid,cleavage,456)","legendgroup":"(mid,cleavage,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-2.3935787342086394],"y":[22.479729503933235],"text":"Stage: cleavage<br>Leachate: mid<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(mid,cleavage,789)","legendgroup":"(mid,cleavage,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-11.981209682678791],"y":[24.825127648884042],"text":"Stage: cleavage<br>Leachate: mid<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(mid,cleavage,101112)","legendgroup":"(mid,cleavage,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-25.442630037800527],"y":[20.486053294335413],"text":"Stage: cleavage<br>Leachate: mid<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(mid,cleavage,131415)","legendgroup":"(mid,cleavage,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-6.6822653860449837],"y":[-0.51064189136776195],"text":"Stage: prawnchip<br>Leachate: mid<br>Parent: 89","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,193,167,1)"}},"hoveron":"points","name":"(mid,prawnchip,89)","legendgroup":"(mid,prawnchip,89)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[44.986471820529211],"y":[-10.128149370111256],"text":"Stage: prawnchip<br>Leachate: mid<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(mid,prawnchip,123)","legendgroup":"(mid,prawnchip,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[27.214486715144762],"y":[-3.0586437170526235],"text":"Stage: prawnchip<br>Leachate: mid<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(mid,prawnchip,456)","legendgroup":"(mid,prawnchip,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-23.098583814986441],"y":[9.7809033651442796],"text":"Stage: prawnchip<br>Leachate: mid<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(mid,prawnchip,101112)","legendgroup":"(mid,prawnchip,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-25.559130962747872],"y":[9.7076248163583347],"text":"Stage: prawnchip<br>Leachate: mid<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(mid,prawnchip,131415)","legendgroup":"(mid,prawnchip,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[41.44610865568837],"y":[-17.549530859774826],"text":"Stage: earlygastrula<br>Leachate: mid<br>Parent: 13","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(174,162,0,1)"}},"hoveron":"points","name":"(mid,earlygastrula,13)","legendgroup":"(mid,earlygastrula,13)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[14.520081036468627],"y":[-27.953700242432085],"text":"Stage: earlygastrula<br>Leachate: mid<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(mid,earlygastrula,456)","legendgroup":"(mid,earlygastrula,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-13.666419120526205],"y":[-24.479671398978965],"text":"Stage: earlygastrula<br>Leachate: mid<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(mid,earlygastrula,789)","legendgroup":"(mid,earlygastrula,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-34.835825262367102],"y":[-8.4466584793003978],"text":"Stage: earlygastrula<br>Leachate: mid<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(mid,earlygastrula,101112)","legendgroup":"(mid,earlygastrula,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-36.463177937841408],"y":[-9.7962212991286304],"text":"Stage: earlygastrula<br>Leachate: mid<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"diamond","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(mid,earlygastrula,131415)","legendgroup":"(mid,earlygastrula,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[46.905759696265171],"y":[26.712155136036039],"text":"Stage: cleavage<br>Leachate: high<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(high,cleavage,123)","legendgroup":"(high,cleavage,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[29.354066717456014],"y":[25.913154489852385],"text":"Stage: cleavage<br>Leachate: high<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(high,cleavage,456)","legendgroup":"(high,cleavage,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.4246123029974726],"y":[21.562274777955089],"text":"Stage: cleavage<br>Leachate: high<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(high,cleavage,789)","legendgroup":"(high,cleavage,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-9.7630773470594647],"y":[26.018961432264611],"text":"Stage: cleavage<br>Leachate: high<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(high,cleavage,101112)","legendgroup":"(high,cleavage,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-20.977287070596645],"y":[23.305792449716566],"text":"Stage: cleavage<br>Leachate: high<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(25,25,25,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(high,cleavage,131415)","legendgroup":"(high,cleavage,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[45.203645816904903],"y":[-5.9103154817412697],"text":"Stage: prawnchip<br>Leachate: high<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(high,prawnchip,123)","legendgroup":"(high,prawnchip,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[26.542512326716366],"y":[-9.8268674782444965],"text":"Stage: prawnchip<br>Leachate: high<br>Parent: 456","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(0,166,255,1)"}},"hoveron":"points","name":"(high,prawnchip,456)","legendgroup":"(high,prawnchip,456)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-12.724884325806597],"y":[-5.6011951313178328],"text":"Stage: prawnchip<br>Leachate: high<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(high,prawnchip,789)","legendgroup":"(high,prawnchip,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-29.917342581515189],"y":[0.29509925443867702],"text":"Stage: prawnchip<br>Leachate: high<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(high,prawnchip,101112)","legendgroup":"(high,prawnchip,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-31.218178929247745],"y":[6.2070093245568509],"text":"Stage: prawnchip<br>Leachate: high<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(187,187,187,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(high,prawnchip,131415)","legendgroup":"(high,prawnchip,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[33.685435254719735],"y":[-24.733505759338446],"text":"Stage: earlygastrula<br>Leachate: high<br>Parent: 45","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(100,178,0,1)"}},"hoveron":"points","name":"(high,earlygastrula,45)","legendgroup":"(high,earlygastrula,45)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[30.999827831458621],"y":[-25.37698340850287],"text":"Stage: earlygastrula<br>Leachate: high<br>Parent: 123","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(0,186,222,1)"}},"hoveron":"points","name":"(high,earlygastrula,123)","legendgroup":"(high,earlygastrula,123)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-16.008038188759926],"y":[-26.648151232778154],"text":"Stage: earlygastrula<br>Leachate: high<br>Parent: 789","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(179,133,255,1)"}},"hoveron":"points","name":"(high,earlygastrula,789)","legendgroup":"(high,earlygastrula,789)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-37.744144077637586],"y":[-11.2643317571133],"text":"Stage: earlygastrula<br>Leachate: high<br>Parent: 101112","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(239,103,235,1)"}},"hoveron":"points","name":"(high,earlygastrula,101112)","legendgroup":"(high,earlygastrula,101112)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-37.65555528893065],"y":[-11.407983487443689],"text":"Stage: earlygastrula<br>Leachate: high<br>Parent: 131415","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,255,255,1)","opacity":0.80000000000000004,"size":13.228346456692915,"symbol":"triangle-up","line":{"width":5.6692913385826778,"color":"rgba(255,99,182,1)"}},"hoveron":"points","name":"(high,earlygastrula,131415)","legendgroup":"(high,earlygastrula,131415)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":40.840182648401829,"r":7.3059360730593621,"b":37.260273972602747,"l":43.105022831050235},"paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"title":{"text":"PCA of all OTUs by Stage, Leachate, and Parental cross","font":{"color":"rgba(0,0,0,1)","family":"","size":17.534246575342465},"x":0.5,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-44.568674995066075,56.463500609659981],"tickmode":"array","ticktext":["-25","0","25","50"],"tickvals":[-25,0,25,50],"categoryorder":"array","categoryarray":["-25","0","25","50"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"PC1 (31.3% variance)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-33.878957162742878,29.597446197882654],"tickmode":"array","ticktext":["-30","-20","-10","0","10","20"],"tickvals":[-30,-20,-10,0,10,20],"categoryorder":"array","categoryarray":["-30","-20","-10","0","10","20"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"PC2 (10.58% variance)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","layer":"below","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"Leachate<br />Stage<br />Parental cross","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3d0c31c82174":{"x":{},"y":{},"fill":{},"colour":{},"shape":{},"text":{},"type":"scatter"}},"cur_data":"3d0c31c82174","visdat":{"3d0c31c82174":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
</section>
<section id="permanova" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> PERMANOVA</h1>
<p>PERMANOVA (Permutational Multivariate Analysis of Variance) on the Aitchison distance matrix to test for differences in microbial composition across experimental factors.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Learn More About PERMANOVA
</div>
</div>
<div class="callout-body-container callout-body">
<p>For a comprehensive explanation of PERMANOVA, including: - How PERMANOVA works and its underlying principles - What inputs are required and what outputs mean - How to interpret results in microbiome context - Complete working examples with all distance metrics</p>
<p>See the dedicated documentation files: - <strong><a href="PERMANOVA_README.md"><code>PERMANOVA_README.md</code></a></strong> - Comprehensive theoretical guide - <strong><a href="permanova_analysis.qmd"><code>permanova_analysis.qmd</code></a></strong> - Working code examples and demonstrations</p>
</div>
</div>
<p>strata are groups within which to constrain permutations. Using strata = parents controls for parent by preventing shuffles across parents; this removes parent as a source of spurious differences.</p>
<p>Caveats to keep in mind</p>
<p>Dispersion sensitivity: PERMANOVA can flag significance if groups have different dispersions (spread), not just different centroids. Consider checking with <code>betadisper</code> and <code>permutest</code>.</p>
<p>Unbalanced designs / small strata: With blocking, you need enough replicates within each parent and across factor levels to allow permutations; otherwise p-values become conservative or unavailable.</p>
<p>Order dependence with by=“terms”: If you care about marginal effects “as if added last,” refit with by=“margin”(or reorder terms thoughtfully).</p>
<section id="fit-betadispr" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="fit-betadispr"><span class="header-section-number">7.1</span> Fit betadispr</h2>
<p><code>betadisper</code> tests whether the within-group multivariate dispersion (spread) differs among groups. It’s the multivariate analogue of Levene’s test: it compares the distances of samples to their group centroid.</p>
<p>What happens under the hood: - Performs a PCoA on your distance matrix. - Computes each group’s centroid in that ordination space. - Calculates each sample’s distance to its group centroid. - Stores those distances for downstream tests/plots.</p>
<p>Define groups to test</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>grp_leachate <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>leachate</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>grp_stage    <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>stage     </span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>grp_int    <span class="ot">&lt;-</span> <span class="fu">interaction</span>(meta_filt<span class="sc">$</span>leachate, meta_filt<span class="sc">$</span>stage)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>grp_night <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>spawn_night</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="night" class="level3" data-number="7.1.1">
<h3 data-number="7.1.1" class="anchored" data-anchor-id="night"><span class="header-section-number">7.1.1</span> night</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit beta dispersion model </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>bd_night <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_night, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with no blocking</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>perm_night <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_night, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Result</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>perm_night</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq Mean Sq      F N.Perm Pr(&gt;F)    
Groups     2  634.07  317.04 6.2856    999  0.001 ***
Residuals 60 3026.32   50.44                         
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>multivariate dispersion differs across spawn nights (groups = nights). In other words, samples from some nights are more spread out around their night-specific centroid than others.</p>
<p>betadisper with group = spawn_night significant heterogeneous dispersion among nights.If your primary tests are leachate/stage, treat night as a nuisance: block by night in PERMANOVA (strata = spawn_night) so permutations happen within each night. This keeps night-to-night variance/dispersion from inflating your p-values for treatment effects.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_night, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="leachate" class="level3" data-number="7.1.2">
<h3 data-number="7.1.2" class="anchored" data-anchor-id="leachate"><span class="header-section-number">7.1.2</span> leachate</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>bd_leachate <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_leachate, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>perm_leachate <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_leachate, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night)  <span class="co"># optional</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>perm_leachate</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df Sum Sq Mean Sq      F N.Perm Pr(&gt;F)
Groups     3   22.5   7.484 0.1393    999  0.932
Residuals 59 3169.8  53.725                     </code></pre>
</div>
</div>
<p>What this tests: Null hypothesis (Ho): all groups have equal average distance-to-centroid (equal dispersion/variance).If p &lt; 0.05, at least one group is more/less dispersed than another.Why use the permutation test? The distances depend on the PCoA, so permutation is safer than the plain ANOVA F-test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_leachate, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="stage" class="level3" data-number="7.1.3">
<h3 data-number="7.1.3" class="anchored" data-anchor-id="stage"><span class="header-section-number">7.1.3</span> stage</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>bd_stage <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_stage, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>perm_stage <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_stage, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>                  <span class="co">#strata = meta_filt$spawn_night)  # optional</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>perm_stage</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq Mean Sq      F N.Perm Pr(&gt;F)
Groups     2   99.76  49.879 1.0498    999  0.368
Residuals 60 2850.65  47.511                     </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_stage, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="leachate-x-stage" class="level3" data-number="7.1.4">
<h3 data-number="7.1.4" class="anchored" data-anchor-id="leachate-x-stage"><span class="header-section-number">7.1.4</span> leachate x stage</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>bd_int <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_int, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>perm_int <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_int, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>                  <span class="co">#strata = meta_filt$spawn_night)  # optional</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>perm_int</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq Mean Sq      F N.Perm Pr(&gt;F)
Groups    11  212.66  19.333 0.3491    999  0.973
Residuals 51 2824.45  55.381                     </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_int, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="compositional_analysis_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="perform-permanova" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="perform-permanova"><span class="header-section-number">7.2</span> Perform PERMANOVA</h2>
<p>PERMANOVA mainly tests centroid location differences among groups (i.e., group separation).</p>
<p>betadisper tests dispersion (variance) differences among groups.</p>
<p>If PERMANOVA is significant and betadisper is non-significant, the signal likely reflects location (good).</p>
<p>If both are significant, some of the PERMANOVA “difference” could be due to dispersion heterogeneity (groups with different spreads). Report both, and interpret cautiously.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PERMANOVA on Aitchison distance</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Test effects of leachate, stage, and their interaction (blocking permutations by parental cross)</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>permanova_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>  aitchison_dist <span class="sc">~</span> leachate <span class="sc">*</span> stage,</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> meta_filt,</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"terms"</span>,</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(permanova_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Blocks:  strata 
Permutation: free
Number of permutations: 999

adonis2(formula = aitchison_dist ~ leachate * stage, data = meta_filt, permutations = 999, by = "terms", strata = meta_filt$spawn_night)
               Df SumOfSqs      R2      F Pr(&gt;F)    
leachate        3     7663 0.03901 0.8356  0.308    
stage           2    23451 0.11936 3.8353  0.001 ***
leachate:stage  6     9434 0.04802 0.5143  1.000    
Residual       51   155919 0.79361                  
Total          62   196468 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save results</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(permanova_result), </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>          <span class="st">"../output/permanova/permanova_result_interactionblocked.csv"</span>,</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Additional PERMANOVA treating stage as continuous time </span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>permanova_cont.time <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  aitchison_dist <span class="sc">~</span> leachate <span class="sc">*</span> hpf,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> meta_filt,</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"terms"</span>,</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night <span class="co"># control for parental cross</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(permanova_cont.time)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Blocks:  strata 
Permutation: free
Number of permutations: 999

adonis2(formula = aitchison_dist ~ leachate * hpf, data = meta_filt, permutations = 999, by = "terms", strata = meta_filt$spawn_night)
             Df SumOfSqs      R2      F Pr(&gt;F)    
leachate      3     7663 0.03901 0.8614  0.285    
hpf           1    20875 0.10625 7.0393  0.001 ***
leachate:hpf  3     4827 0.02457 0.5426  0.996    
Residual     55   163102 0.83017                  
Total        62   196468 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save results</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(permanova_cont.time), </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>          <span class="st">"../output/permanova/permanova_result_cont.time.csv"</span>,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="summary" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Summary</h1>
<p>This analysis implemented the key recommendations from Gloor et al.&nbsp;2017 for compositional microbiome data analysis:</p>
<ol type="1">
<li><strong>CLR Transformation</strong>: Applied centered log-ratio transformation to convert compositional data to Euclidean space</li>
<li><strong>Aitchison Distance</strong>: Calculated Euclidean distance on CLR-transformed data as the appropriate dissimilarity metric</li>
<li><strong>PCA Biplot</strong>: Created variance-based PCA visualizations on CLR-transformed data</li>
<li><strong>PERMANOVA</strong>: Tested for significant differences in microbial composition across experimental factors</li>
</ol>
<section id="key-findings" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="key-findings"><span class="header-section-number">8.1</span> Key Findings</h2>
<ul>
<li>PCA plots visually show clustering patterns by Developmental stage (aka timepoint/ hours post fertilization), Night of Spawn, and Parental cross (the later two are confounded)</li>
<li>PERMANOVA results (blocked across Night of Spawn) indicate significant effects of Developmental stage (aka timepoint/ hours post fertilization), and non-significance for leachate and leahcate:stage interactions</li>
</ul>
</section>
<section id="next-steps" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="next-steps"><span class="header-section-number">8.2</span> Next Steps</h2>
<ul>
<li>Differential abundance analysis (MaAslin2)</li>
<li>Conduct PERMANOVA on differentially abundant taxa</li>
<li>Functional annotation of differentially abundant taxa (picrust)</li>
<li>Differential abundance of functions (functional enrichment) (MaAslin2)</li>
</ul>
</section>
<section id="output-files" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="output-files"><span class="header-section-number">8.3</span> Output Files</h2>
<p>All results and visualizations have been saved to the <code>../output/permanova</code> directory: - PCA biplots - PERMANOVA results tables</p>
</section>
</section>
<section id="session-info" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Session Info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] plotly_4.11.0         ANCOMBC_2.10.1        corrplot_0.95        
 [4] reshape2_1.4.4        zCompositions_1.5.0-5 survival_3.8-3       
 [7] truncnorm_1.0-9       MASS_7.3-65           compositions_2.0-9   
[10] vegan_2.7-2           permute_0.9-8         phyloseq_1.52.0      
[13] qiime2R_0.99.6        lubridate_1.9.4       forcats_1.0.1        
[16] stringr_1.5.2         dplyr_1.1.4           purrr_1.1.0          
[19] readr_2.1.5           tidyr_1.3.1           tibble_3.3.0         
[22] ggplot2_4.0.0         tidyverse_2.0.0      

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      tensorA_0.36.2.1        rstudioapi_0.17.1      
  [4] jsonlite_2.0.0          magrittr_2.0.4          TH.data_1.1-4          
  [7] nloptr_2.2.1            farver_2.1.2            rmarkdown_2.30         
 [10] ragg_1.5.0              fs_1.6.6                vctrs_0.6.5            
 [13] multtest_2.64.0         minqa_1.2.8             base64enc_0.1-3        
 [16] htmltools_0.5.8.1       energy_1.7-12           haven_2.5.5            
 [19] cellranger_1.1.0        Rhdf5lib_1.30.0         Formula_1.2-5          
 [22] rhdf5_2.52.1            htmlwidgets_1.6.4       sandwich_3.1-1         
 [25] plyr_1.8.9              zoo_1.8-14              rootSolve_1.8.2.4      
 [28] igraph_2.2.0            lifecycle_1.0.4         iterators_1.0.14       
 [31] pkgconfig_2.0.3         Matrix_1.7-4            R6_2.6.1               
 [34] fastmap_1.2.0           GenomeInfoDbData_1.2.14 rbibutils_2.3          
 [37] numDeriv_2016.8-1.1     digest_0.6.37           Exact_3.3              
 [40] colorspace_2.1-2        S4Vectors_0.46.0        crosstalk_1.2.2        
 [43] textshaping_1.0.4       Hmisc_5.2-4             labeling_0.4.3         
 [46] timechange_0.3.0        httr_1.4.7              mgcv_1.9-3             
 [49] compiler_4.5.1          rngtools_1.5.2          proxy_0.4-27           
 [52] gsl_2.1-8               bit64_4.6.0-1           withr_3.0.2            
 [55] doParallel_1.0.17       htmlTable_2.4.3         S7_0.2.0               
 [58] backports_1.5.0         bayesm_3.1-6            biomformat_1.36.0      
 [61] gtools_3.9.5            CVXR_1.0-15             gld_2.6.8              
 [64] tools_4.5.1             foreign_0.8-90          ape_5.8-1              
 [67] nnet_7.3-20             glue_1.8.0              nlme_3.1-168           
 [70] rhdf5filters_1.20.0     grid_4.5.1              checkmate_2.3.3        
 [73] cluster_2.1.8.1         ade4_1.7-23             generics_0.1.4         
 [76] gtable_0.3.6            tzdb_0.5.0              class_7.3-23           
 [79] data.table_1.17.8       lmom_3.2                hms_1.1.4              
 [82] XVector_0.48.0          BiocGenerics_0.54.1     foreach_1.5.2          
 [85] pillar_1.11.1           vroom_1.6.6             robustbase_0.99-6      
 [88] splines_4.5.1           lattice_0.22-7          gmp_0.7-5              
 [91] bit_4.6.0               tidyselect_1.2.1        Biostrings_2.76.0      
 [94] knitr_1.50              reformulas_0.4.1        gridExtra_2.3          
 [97] IRanges_2.42.0          stats4_4.5.1            xfun_0.53              
[100] expm_1.0-0              Biobase_2.68.0          DEoptimR_1.1-4         
[103] DT_0.34.0               stringi_1.8.7           UCSC.utils_1.4.0       
[106] lazyeval_0.2.2          yaml_2.3.10             boot_1.3-32            
[109] evaluate_1.0.5          codetools_0.2-20        cli_3.6.5              
[112] rpart_4.1.24            systemfonts_1.3.1       DescTools_0.99.60      
[115] Rdpack_2.6.4            dichromat_2.0-0.1       Rcpp_1.1.0             
[118] GenomeInfoDb_1.44.3     readxl_1.4.5            parallel_4.5.1         
[121] doRNG_1.8.6.2           lme4_1.1-37             Rmpfr_1.1-1            
[124] viridisLite_0.4.2       mvtnorm_1.3-3           lmerTest_3.1-3         
[127] scales_1.4.0            e1071_1.7-16            crayon_1.5.3           
[130] rlang_1.1.6             multcomp_1.4-29        </code></pre>
</div>
</div>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb50" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Compositional Data Analysis" </span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Following Gloor et al. 2017 recommendations for microbiome compositional data analysis"</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Sarah Tanja" </span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 10/17/2025 </span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="an">date-modified:</span><span class="co"> today</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="co">  gfm: </span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true </span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true </span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="co">  html: </span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: journal</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github </span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full </span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-background: true</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: </span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="co">      source: true </span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="co">      toggle: true</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true </span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 2</span></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: left </span></span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true </span></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: kable </span></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="co">    smooth-scroll: true </span></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a><span class="co">    link-external-icon: true </span></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="co">    link-external-newwindow: true</span></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> "../microbiome_bibtex.bib"</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a><span class="an">reference-location:</span><span class="co"> margin</span></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="an">citation-location:</span><span class="co"> margin</span></span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>,         <span class="co"># Display code chunks</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>,        <span class="co"># Evaluate code chunks</span></span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>,     <span class="co"># Hide warnings</span></span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,     <span class="co"># Hide messages</span></span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">comment =</span> <span class="st">""</span>         <span class="co"># Prevents appending '##' to beginning of lines in code output</span></span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># Background</span></span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a>This analysis follows the recommendations from Gloor et al. 2017 for compositional data analysis of microbiome data. Microbiome sequence count data are compositional - they represent relative abundances constrained to sum to a total (sequencing depth). This requires specialized statistical methods that account for the compositional nature of the data.</span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a>Key recommendations from Gloor et al. 2017:</span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Use centered log-ratio (CLR) transformation to convert compositional data to Euclidean space</span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Calculate Aitchison distance (Euclidean distance on CLR-transformed data) for dissimilarity</span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Use variance-based methods like PCA on CLR-transformed data</span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Perform statistical tests on transformed data</span>
<span id="cb50-50"><a href="#cb50-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-51"><a href="#cb50-51" aria-hidden="true" tabindex="-1"></a><span class="fu"># Setup</span></span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-55"><a href="#cb50-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file paths</span></span>
<span id="cb50-56"><a href="#cb50-56" aria-hidden="true" tabindex="-1"></a>feature_table_path <span class="ot">&lt;-</span> <span class="st">"../salipante/241121_StonyCoral/270x200/250414_StonyCoral_270x200_featureTable_filtered.qza"</span></span>
<span id="cb50-57"><a href="#cb50-57" aria-hidden="true" tabindex="-1"></a>metadata_path <span class="ot">&lt;-</span> <span class="st">"../metadata/meta.csv"</span></span>
<span id="cb50-58"><a href="#cb50-58" aria-hidden="true" tabindex="-1"></a>taxonomy_path <span class="ot">&lt;-</span> <span class="st">"../salipante/241121_StonyCoral/270x200/250414_270x200_representative-sequences_taxonomy.qza"</span></span>
<span id="cb50-59"><a href="#cb50-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-60"><a href="#cb50-60" aria-hidden="true" tabindex="-1"></a><span class="fu">## Install &amp; load packages</span></span>
<span id="cb50-61"><a href="#cb50-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-62"><a href="#cb50-62" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb50-63"><a href="#cb50-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Check R version</span></span>
<span id="cb50-64"><a href="#cb50-64" aria-hidden="true" tabindex="-1"></a><span class="co"># R.version.string</span></span>
<span id="cb50-65"><a href="#cb50-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-66"><a href="#cb50-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Install and load required packages</span></span>
<span id="cb50-67"><a href="#cb50-67" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("BiocManager", quietly = TRUE)) {</span></span>
<span id="cb50-68"><a href="#cb50-68" aria-hidden="true" tabindex="-1"></a><span class="co">#  install.packages("BiocManager")</span></span>
<span id="cb50-69"><a href="#cb50-69" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-70"><a href="#cb50-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-71"><a href="#cb50-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Install Bioconductor packages</span></span>
<span id="cb50-72"><a href="#cb50-72" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("phyloseq", quietly = TRUE)) {</span></span>
<span id="cb50-73"><a href="#cb50-73" aria-hidden="true" tabindex="-1"></a><span class="co">#  BiocManager::install("phyloseq")</span></span>
<span id="cb50-74"><a href="#cb50-74" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-75"><a href="#cb50-75" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("ANCOMBC", quietly = TRUE)) {</span></span>
<span id="cb50-76"><a href="#cb50-76" aria-hidden="true" tabindex="-1"></a><span class="co">#  BiocManager::install("ANCOMBC")</span></span>
<span id="cb50-77"><a href="#cb50-77" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-78"><a href="#cb50-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-79"><a href="#cb50-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Install CRAN packages</span></span>
<span id="cb50-80"><a href="#cb50-80" aria-hidden="true" tabindex="-1"></a><span class="co">#required_packages &lt;- c("tidyverse", "vegan", "compositions", "zCompositions", </span></span>
<span id="cb50-81"><a href="#cb50-81" aria-hidden="true" tabindex="-1"></a><span class="co">#                       "ggplot2", "reshape2", "corrplot", "devtools")</span></span>
<span id="cb50-82"><a href="#cb50-82" aria-hidden="true" tabindex="-1"></a><span class="co">#for (pkg in required_packages) {</span></span>
<span id="cb50-83"><a href="#cb50-83" aria-hidden="true" tabindex="-1"></a><span class="co">#  if (!requireNamespace(pkg, quietly = TRUE)) {</span></span>
<span id="cb50-84"><a href="#cb50-84" aria-hidden="true" tabindex="-1"></a><span class="co">#    install.packages(pkg)</span></span>
<span id="cb50-85"><a href="#cb50-85" aria-hidden="true" tabindex="-1"></a><span class="co">#  }</span></span>
<span id="cb50-86"><a href="#cb50-86" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-87"><a href="#cb50-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-88"><a href="#cb50-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Install qiime2R from GitHub</span></span>
<span id="cb50-89"><a href="#cb50-89" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("qiime2R", quietly = TRUE)) {</span></span>
<span id="cb50-90"><a href="#cb50-90" aria-hidden="true" tabindex="-1"></a><span class="co">#  devtools::install_github("jbisanz/qiime2R")</span></span>
<span id="cb50-91"><a href="#cb50-91" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-92"><a href="#cb50-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-93"><a href="#cb50-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Install SpiecEasi for SparCC (optional, can be slow to install)</span></span>
<span id="cb50-94"><a href="#cb50-94" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!requireNamespace("SpiecEasi", quietly = TRUE)) {</span></span>
<span id="cb50-95"><a href="#cb50-95" aria-hidden="true" tabindex="-1"></a><span class="co">#  devtools::install_github("zdk123/SpiecEasi")</span></span>
<span id="cb50-96"><a href="#cb50-96" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span>
<span id="cb50-97"><a href="#cb50-97" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-98"><a href="#cb50-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-101"><a href="#cb50-101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-102"><a href="#cb50-102" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb50-103"><a href="#cb50-103" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb50-104"><a href="#cb50-104" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb50-105"><a href="#cb50-105" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb50-106"><a href="#cb50-106" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb50-107"><a href="#cb50-107" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb50-108"><a href="#cb50-108" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zCompositions)</span>
<span id="cb50-109"><a href="#cb50-109" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb50-110"><a href="#cb50-110" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb50-111"><a href="#cb50-111" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb50-112"><a href="#cb50-112" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb50-113"><a href="#cb50-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-114"><a href="#cb50-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-115"><a href="#cb50-115" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load Data</span></span>
<span id="cb50-116"><a href="#cb50-116" aria-hidden="true" tabindex="-1"></a><span class="fu">## Load metadata</span></span>
<span id="cb50-119"><a href="#cb50-119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-120"><a href="#cb50-120" aria-hidden="true" tabindex="-1"></a><span class="co"># Load metadata</span></span>
<span id="cb50-121"><a href="#cb50-121" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(metadata_path)</span>
<span id="cb50-122"><a href="#cb50-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-123"><a href="#cb50-123" aria-hidden="true" tabindex="-1"></a><span class="co"># set factors</span></span>
<span id="cb50-124"><a href="#cb50-124" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb50-125"><a href="#cb50-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb50-126"><a href="#cb50-126" aria-hidden="true" tabindex="-1"></a>    <span class="at">collection_date =</span> <span class="fu">as.Date</span>(collection_date, <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb50-127"><a href="#cb50-127" aria-hidden="true" tabindex="-1"></a>    <span class="at">stage    =</span> <span class="fu">factor</span>(stage,    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"cleavage"</span>, <span class="st">"prawnchip"</span>, <span class="st">"earlygastrula"</span>), <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-128"><a href="#cb50-128" aria-hidden="true" tabindex="-1"></a>    <span class="at">leachate =</span> <span class="fu">factor</span>(leachate, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"control"</span>, <span class="st">"low"</span>, <span class="st">"mid"</span>, <span class="st">"high"</span>),        <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb50-129"><a href="#cb50-129" aria-hidden="true" tabindex="-1"></a>    <span class="at">spawn_night =</span> <span class="fu">factor</span>(</span>
<span id="cb50-130"><a href="#cb50-130" aria-hidden="true" tabindex="-1"></a>      collection_date,</span>
<span id="cb50-131"><a href="#cb50-131" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels  =</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"06-Jul-2024"</span>, <span class="st">"07-Jul-2024"</span>, <span class="st">"08-Jul-2024"</span>), <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb50-132"><a href="#cb50-132" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels  =</span> <span class="fu">c</span>(<span class="st">"July 6th"</span>, <span class="st">"July 7th"</span>, <span class="st">"July 8th"</span>),</span>
<span id="cb50-133"><a href="#cb50-133" aria-hidden="true" tabindex="-1"></a>      <span class="at">ordered =</span> <span class="cn">TRUE</span></span>
<span id="cb50-134"><a href="#cb50-134" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb50-135"><a href="#cb50-135" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-136"><a href="#cb50-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-137"><a href="#cb50-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-138"><a href="#cb50-138" aria-hidden="true" tabindex="-1"></a><span class="co"># View metadata structure</span></span>
<span id="cb50-139"><a href="#cb50-139" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span>
<span id="cb50-140"><a href="#cb50-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-141"><a href="#cb50-141" aria-hidden="true" tabindex="-1"></a><span class="fu">## Import QIIME2 artifacts</span></span>
<span id="cb50-142"><a href="#cb50-142" aria-hidden="true" tabindex="-1"></a><span class="fu">### Feature table</span></span>
<span id="cb50-145"><a href="#cb50-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-146"><a href="#cb50-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Load feature table from QIIME2 artifact</span></span>
<span id="cb50-147"><a href="#cb50-147" aria-hidden="true" tabindex="-1"></a>feature_table <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(feature_table_path)<span class="sc">$</span>data</span>
<span id="cb50-148"><a href="#cb50-148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-149"><a href="#cb50-149" aria-hidden="true" tabindex="-1"></a><span class="fu">### Taxonomy</span></span>
<span id="cb50-152"><a href="#cb50-152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-153"><a href="#cb50-153" aria-hidden="true" tabindex="-1"></a><span class="co"># Load taxonomy (optional, for annotation)</span></span>
<span id="cb50-154"><a href="#cb50-154" aria-hidden="true" tabindex="-1"></a>taxonomy <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(taxonomy_path)<span class="sc">$</span>data</span>
<span id="cb50-155"><a href="#cb50-155" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-156"><a href="#cb50-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-159"><a href="#cb50-159" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-160"><a href="#cb50-160" aria-hidden="true" tabindex="-1"></a><span class="co"># Display data dimensions</span></span>
<span id="cb50-161"><a href="#cb50-161" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Feature table dimensions:"</span>, <span class="fu">dim</span>(feature_table), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-162"><a href="#cb50-162" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of samples in metadata:"</span>, <span class="fu">nrow</span>(meta), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-163"><a href="#cb50-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-164"><a href="#cb50-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-165"><a href="#cb50-165" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prepare data for analysis</span></span>
<span id="cb50-166"><a href="#cb50-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-169"><a href="#cb50-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-170"><a href="#cb50-170" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert feature table to matrix (samples as rows, features as columns)</span></span>
<span id="cb50-171"><a href="#cb50-171" aria-hidden="true" tabindex="-1"></a><span class="co"># QIIME2 format typically has features as rows, samples as columns</span></span>
<span id="cb50-172"><a href="#cb50-172" aria-hidden="true" tabindex="-1"></a>feature_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(feature_table)</span>
<span id="cb50-173"><a href="#cb50-173" aria-hidden="true" tabindex="-1"></a>feature_matrix_t <span class="ot">&lt;-</span> <span class="fu">t</span>(feature_matrix)  <span class="co"># Transpose so samples are rows</span></span>
<span id="cb50-174"><a href="#cb50-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-175"><a href="#cb50-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure sample IDs match between feature table and metadata</span></span>
<span id="cb50-176"><a href="#cb50-176" aria-hidden="true" tabindex="-1"></a>sample_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(feature_matrix_t)</span>
<span id="cb50-177"><a href="#cb50-177" aria-hidden="true" tabindex="-1"></a>meta_filt <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span></span>
<span id="cb50-178"><a href="#cb50-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">%in%</span> sample_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-179"><a href="#cb50-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(sample_id, sample_ids))</span>
<span id="cb50-180"><a href="#cb50-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-181"><a href="#cb50-181" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Samples in feature table:"</span>, <span class="fu">nrow</span>(feature_matrix_t), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-182"><a href="#cb50-182" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Samples in metadata:"</span>, <span class="fu">nrow</span>(meta_filt), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-183"><a href="#cb50-183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-184"><a href="#cb50-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-185"><a href="#cb50-185" aria-hidden="true" tabindex="-1"></a><span class="fu"># CLR Transformation</span></span>
<span id="cb50-186"><a href="#cb50-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-187"><a href="#cb50-187" aria-hidden="true" tabindex="-1"></a>The centered log-ratio (CLR) transformation converts compositional data to Euclidean space. It handles zeros by replacing them with small values (geometric Bayesian multiplicative replacement).</span>
<span id="cb50-188"><a href="#cb50-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-191"><a href="#cb50-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-192"><a href="#cb50-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace zeros with small values using cmultRepl from zCompositions</span></span>
<span id="cb50-193"><a href="#cb50-193" aria-hidden="true" tabindex="-1"></a><span class="co"># This is necessary because log transformation is undefined for zeros</span></span>
<span id="cb50-194"><a href="#cb50-194" aria-hidden="true" tabindex="-1"></a>feature_matrix_nozero <span class="ot">&lt;-</span> <span class="fu">cmultRepl</span>(feature_matrix_t, <span class="at">method =</span> <span class="st">"CZM"</span>, <span class="at">label =</span> <span class="dv">0</span>)</span>
<span id="cb50-195"><a href="#cb50-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-196"><a href="#cb50-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply CLR transformation</span></span>
<span id="cb50-197"><a href="#cb50-197" aria-hidden="true" tabindex="-1"></a>feature_clr <span class="ot">&lt;-</span> <span class="fu">clr</span>(feature_matrix_nozero)</span>
<span id="cb50-198"><a href="#cb50-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-199"><a href="#cb50-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame for easier handling</span></span>
<span id="cb50-200"><a href="#cb50-200" aria-hidden="true" tabindex="-1"></a>feature_clr_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(feature_clr)</span>
<span id="cb50-201"><a href="#cb50-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-202"><a href="#cb50-202" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"CLR-transformed data dimensions:"</span>, <span class="fu">dim</span>(feature_clr_df), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-203"><a href="#cb50-203" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_clr_df[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb50-204"><a href="#cb50-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-205"><a href="#cb50-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-206"><a href="#cb50-206" aria-hidden="true" tabindex="-1"></a><span class="fu"># Aitchison Distance</span></span>
<span id="cb50-207"><a href="#cb50-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-208"><a href="#cb50-208" aria-hidden="true" tabindex="-1"></a>The Aitchison distance is the Euclidean distance between samples after CLR transformation. This is the appropriate distance metric for compositional data.</span>
<span id="cb50-209"><a href="#cb50-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-212"><a href="#cb50-212" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-213"><a href="#cb50-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Aitchison distance (Euclidean distance on CLR-transformed data)</span></span>
<span id="cb50-214"><a href="#cb50-214" aria-hidden="true" tabindex="-1"></a>aitchison_dist <span class="ot">&lt;-</span> <span class="fu">dist</span>(feature_clr, <span class="at">method =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb50-215"><a href="#cb50-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-216"><a href="#cb50-216" aria-hidden="true" tabindex="-1"></a><span class="co"># Display distance matrix summary</span></span>
<span id="cb50-217"><a href="#cb50-217" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Aitchison distance matrix summary:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-218"><a href="#cb50-218" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">as.vector</span>(aitchison_dist))</span>
<span id="cb50-219"><a href="#cb50-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-220"><a href="#cb50-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-221"><a href="#cb50-221" aria-hidden="true" tabindex="-1"></a><span class="fu"># PCA Biplots</span></span>
<span id="cb50-222"><a href="#cb50-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-223"><a href="#cb50-223" aria-hidden="true" tabindex="-1"></a>Variance-based compositional PCA on CLR-transformed data.</span>
<span id="cb50-224"><a href="#cb50-224" aria-hidden="true" tabindex="-1"></a><span class="fu">## Perform PCA</span></span>
<span id="cb50-227"><a href="#cb50-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-228"><a href="#cb50-228" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on CLR-transformed data</span></span>
<span id="cb50-229"><a href="#cb50-229" aria-hidden="true" tabindex="-1"></a>pca_result <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(feature_clr, <span class="at">scale. =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-230"><a href="#cb50-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-231"><a href="#cb50-231" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate variance explained</span></span>
<span id="cb50-232"><a href="#cb50-232" aria-hidden="true" tabindex="-1"></a>variance_explained <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_result)<span class="sc">$</span>importance[<span class="dv">2</span>, ] <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb50-233"><a href="#cb50-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-234"><a href="#cb50-234" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame for plotting</span></span>
<span id="cb50-235"><a href="#cb50-235" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb50-236"><a href="#cb50-236" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pca_result<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb50-237"><a href="#cb50-237" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pca_result<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb50-238"><a href="#cb50-238" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample_id =</span> <span class="fu">rownames</span>(feature_clr)</span>
<span id="cb50-239"><a href="#cb50-239" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-240"><a href="#cb50-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-241"><a href="#cb50-241" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with metadata</span></span>
<span id="cb50-242"><a href="#cb50-242" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> pca_df <span class="sc">%&gt;%</span></span>
<span id="cb50-243"><a href="#cb50-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta_filt, <span class="at">by =</span> <span class="st">"sample_id"</span>)</span>
<span id="cb50-244"><a href="#cb50-244" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-245"><a href="#cb50-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-246"><a href="#cb50-246" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plot stage x leachate x cross</span></span>
<span id="cb50-247"><a href="#cb50-247" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=6}</span></span>
<span id="cb50-248"><a href="#cb50-248" aria-hidden="true" tabindex="-1"></a>pca_plot_cross <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb50-249"><a href="#cb50-249" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb50-250"><a href="#cb50-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb50-251"><a href="#cb50-251" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb50-252"><a href="#cb50-252" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                   <span class="co"># inside color</span></span>
<span id="cb50-253"><a href="#cb50-253" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(parents),       <span class="co"># outline color</span></span>
<span id="cb50-254"><a href="#cb50-254" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate                <span class="co"># shape must be 21–25</span></span>
<span id="cb50-255"><a href="#cb50-255" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-256"><a href="#cb50-256" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb50-257"><a href="#cb50-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span>   <span class="co"># stroke controls outline width</span></span>
<span id="cb50-258"><a href="#cb50-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb50-259"><a href="#cb50-259" aria-hidden="true" tabindex="-1"></a>  <span class="co"># light grey -&gt; dark grey for discrete stages (shows in legend)</span></span>
<span id="cb50-260"><a href="#cb50-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-261"><a href="#cb50-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb50-262"><a href="#cb50-262" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Parental cross"</span>,</span>
<span id="cb50-263"><a href="#cb50-263" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-264"><a href="#cb50-264" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-265"><a href="#cb50-265" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb50-266"><a href="#cb50-266" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Parental cross"</span></span>
<span id="cb50-267"><a href="#cb50-267" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb50-268"><a href="#cb50-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb50-269"><a href="#cb50-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb50-270"><a href="#cb50-270" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb50-271"><a href="#cb50-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb50-272"><a href="#cb50-272" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb50-273"><a href="#cb50-273" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3) make the fill legend draw a filled shape with an outline</span></span>
<span id="cb50-274"><a href="#cb50-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb50-275"><a href="#cb50-275" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb50-276"><a href="#cb50-276" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>), </span>
<span id="cb50-277"><a href="#cb50-277" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb50-278"><a href="#cb50-278" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb50-279"><a href="#cb50-279" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb50-280"><a href="#cb50-280" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb50-281"><a href="#cb50-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-282"><a href="#cb50-282" aria-hidden="true" tabindex="-1"></a>pca_plot_cross</span>
<span id="cb50-283"><a href="#cb50-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-284"><a href="#cb50-284" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot</span></span>
<span id="cb50-285"><a href="#cb50-285" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/compositional_pca_cross.png"</span>, pca_plot_cross, </span>
<span id="cb50-286"><a href="#cb50-286" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">7</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb50-287"><a href="#cb50-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-288"><a href="#cb50-288" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plot stage x leachate x spawn night</span></span>
<span id="cb50-289"><a href="#cb50-289" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=6}</span></span>
<span id="cb50-290"><a href="#cb50-290" aria-hidden="true" tabindex="-1"></a>pca_plot_spawn_night <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb50-291"><a href="#cb50-291" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb50-292"><a href="#cb50-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb50-293"><a href="#cb50-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb50-294"><a href="#cb50-294" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                   <span class="co"># inside color</span></span>
<span id="cb50-295"><a href="#cb50-295" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(spawn_night),       <span class="co"># outline color</span></span>
<span id="cb50-296"><a href="#cb50-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate                <span class="co"># shape must be 21–25</span></span>
<span id="cb50-297"><a href="#cb50-297" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-298"><a href="#cb50-298" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb50-299"><a href="#cb50-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span>   <span class="co"># stroke controls outline width</span></span>
<span id="cb50-300"><a href="#cb50-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb50-301"><a href="#cb50-301" aria-hidden="true" tabindex="-1"></a>  <span class="co"># light grey -&gt; dark grey for discrete stages (shows in legend)</span></span>
<span id="cb50-302"><a href="#cb50-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-303"><a href="#cb50-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb50-304"><a href="#cb50-304" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Spawn night"</span>,</span>
<span id="cb50-305"><a href="#cb50-305" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-306"><a href="#cb50-306" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-307"><a href="#cb50-307" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb50-308"><a href="#cb50-308" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Spawn night"</span></span>
<span id="cb50-309"><a href="#cb50-309" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb50-310"><a href="#cb50-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb50-311"><a href="#cb50-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb50-312"><a href="#cb50-312" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb50-313"><a href="#cb50-313" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb50-314"><a href="#cb50-314" aria-hidden="true" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb50-315"><a href="#cb50-315" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3) make the fill legend draw a filled shape with an outline</span></span>
<span id="cb50-316"><a href="#cb50-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb50-317"><a href="#cb50-317" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb50-318"><a href="#cb50-318" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>), </span>
<span id="cb50-319"><a href="#cb50-319" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb50-320"><a href="#cb50-320" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb50-321"><a href="#cb50-321" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb50-322"><a href="#cb50-322" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb50-323"><a href="#cb50-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-324"><a href="#cb50-324" aria-hidden="true" tabindex="-1"></a>pca_plot_spawn_night</span>
<span id="cb50-325"><a href="#cb50-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-326"><a href="#cb50-326" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot</span></span>
<span id="cb50-327"><a href="#cb50-327" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/compositional_pca_spawn_night.png"</span>, pca_plot_spawn_night, </span>
<span id="cb50-328"><a href="#cb50-328" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">7</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb50-329"><a href="#cb50-329" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-330"><a href="#cb50-330" aria-hidden="true" tabindex="-1"></a><span class="fu">## Interactive plotly</span></span>
<span id="cb50-333"><a href="#cb50-333" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-334"><a href="#cb50-334" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb50-335"><a href="#cb50-335" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb50-336"><a href="#cb50-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-337"><a href="#cb50-337" aria-hidden="true" tabindex="-1"></a><span class="co"># add a tooltip aesthetic to your existing ggplot</span></span>
<span id="cb50-338"><a href="#cb50-338" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb50-339"><a href="#cb50-339" aria-hidden="true" tabindex="-1"></a>  pca_df,</span>
<span id="cb50-340"><a href="#cb50-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(</span>
<span id="cb50-341"><a href="#cb50-341" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb50-342"><a href="#cb50-342" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> stage,                    <span class="co"># inside color</span></span>
<span id="cb50-343"><a href="#cb50-343" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">factor</span>(parents),        <span class="co"># outline color</span></span>
<span id="cb50-344"><a href="#cb50-344" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> leachate,                <span class="co"># shape must be 21–25</span></span>
<span id="cb50-345"><a href="#cb50-345" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">paste0</span>(</span>
<span id="cb50-346"><a href="#cb50-346" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Stage: "</span>, stage,</span>
<span id="cb50-347"><a href="#cb50-347" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;br&gt;Leachate: "</span>, leachate,</span>
<span id="cb50-348"><a href="#cb50-348" aria-hidden="true" tabindex="-1"></a>      <span class="st">"&lt;br&gt;Parent: "</span>, parents</span>
<span id="cb50-349"><a href="#cb50-349" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb50-350"><a href="#cb50-350" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-351"><a href="#cb50-351" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb50-352"><a href="#cb50-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-353"><a href="#cb50-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>, <span class="dv">25</span>)[<span class="fu">seq_along</span>(<span class="fu">levels</span>(pca_df<span class="sc">$</span>leachate))]) <span class="sc">+</span></span>
<span id="cb50-354"><a href="#cb50-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grey</span>(<span class="at">name =</span> <span class="st">"Stage"</span>, <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-355"><a href="#cb50-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb50-356"><a href="#cb50-356" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"PCA of all OTUs by Stage, Leachate, and Parental cross"</span>,</span>
<span id="cb50-357"><a href="#cb50-357" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-358"><a href="#cb50-358" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">"% variance)"</span>),</span>
<span id="cb50-359"><a href="#cb50-359" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">"Leachate"</span>,</span>
<span id="cb50-360"><a href="#cb50-360" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Parental cross"</span></span>
<span id="cb50-361"><a href="#cb50-361" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb50-362"><a href="#cb50-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb50-363"><a href="#cb50-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb50-364"><a href="#cb50-364" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb50-365"><a href="#cb50-365" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb50-366"><a href="#cb50-366" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb50-367"><a href="#cb50-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb50-368"><a href="#cb50-368" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape  =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">1</span>),</span>
<span id="cb50-369"><a href="#cb50-369" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill   =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.1</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb50-370"><a href="#cb50-370" aria-hidden="true" tabindex="-1"></a>                          <span class="at">order =</span> <span class="dv">2</span>),</span>
<span id="cb50-371"><a href="#cb50-371" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">order =</span> <span class="dv">3</span>)</span>
<span id="cb50-372"><a href="#cb50-372" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-373"><a href="#cb50-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-374"><a href="#cb50-374" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to plotly and use only the custom text in the tooltip</span></span>
<span id="cb50-375"><a href="#cb50-375" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf_interactive <span class="ot">&lt;-</span> <span class="fu">ggplotly</span>(pca_plot_hpf, <span class="at">tooltip =</span> <span class="st">"text"</span>)</span>
<span id="cb50-376"><a href="#cb50-376" aria-hidden="true" tabindex="-1"></a>pca_plot_hpf_interactive</span>
<span id="cb50-377"><a href="#cb50-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-378"><a href="#cb50-378" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-379"><a href="#cb50-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-380"><a href="#cb50-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-381"><a href="#cb50-381" aria-hidden="true" tabindex="-1"></a><span class="fu"># PERMANOVA</span></span>
<span id="cb50-382"><a href="#cb50-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-383"><a href="#cb50-383" aria-hidden="true" tabindex="-1"></a>PERMANOVA (Permutational Multivariate Analysis of Variance) on the Aitchison distance matrix to test for differences in microbial composition across experimental factors.</span>
<span id="cb50-384"><a href="#cb50-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-385"><a href="#cb50-385" aria-hidden="true" tabindex="-1"></a>::: {.callout-tip}</span>
<span id="cb50-386"><a href="#cb50-386" aria-hidden="true" tabindex="-1"></a><span class="fu">## Learn More About PERMANOVA</span></span>
<span id="cb50-387"><a href="#cb50-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-388"><a href="#cb50-388" aria-hidden="true" tabindex="-1"></a>For a comprehensive explanation of PERMANOVA, including:</span>
<span id="cb50-389"><a href="#cb50-389" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How PERMANOVA works and its underlying principles</span>
<span id="cb50-390"><a href="#cb50-390" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What inputs are required and what outputs mean</span>
<span id="cb50-391"><a href="#cb50-391" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How to interpret results in microbiome context</span>
<span id="cb50-392"><a href="#cb50-392" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Complete working examples with all distance metrics</span>
<span id="cb50-393"><a href="#cb50-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-394"><a href="#cb50-394" aria-hidden="true" tabindex="-1"></a>See the dedicated documentation files:</span>
<span id="cb50-395"><a href="#cb50-395" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**[`PERMANOVA_README.md`](PERMANOVA_README.md)** - Comprehensive theoretical guide</span>
<span id="cb50-396"><a href="#cb50-396" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**[`permanova_analysis.qmd`](permanova_analysis.qmd)** - Working code examples and demonstrations</span>
<span id="cb50-397"><a href="#cb50-397" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb50-398"><a href="#cb50-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-399"><a href="#cb50-399" aria-hidden="true" tabindex="-1"></a>strata are groups within which to constrain permutations. Using </span>
<span id="cb50-400"><a href="#cb50-400" aria-hidden="true" tabindex="-1"></a>strata = parents controls for parent by preventing shuffles across parents; this removes parent as a source of spurious differences.</span>
<span id="cb50-401"><a href="#cb50-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-402"><a href="#cb50-402" aria-hidden="true" tabindex="-1"></a>Caveats to keep in mind</span>
<span id="cb50-403"><a href="#cb50-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-404"><a href="#cb50-404" aria-hidden="true" tabindex="-1"></a>Dispersion sensitivity: PERMANOVA can flag significance if groups have different dispersions (spread), not just different centroids. Consider checking with <span class="in">`betadisper`</span> and <span class="in">`permutest`</span>.</span>
<span id="cb50-405"><a href="#cb50-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-406"><a href="#cb50-406" aria-hidden="true" tabindex="-1"></a>Unbalanced designs / small strata: With blocking, you need enough replicates within each parent and across factor levels to allow permutations; otherwise p-values become conservative or unavailable.</span>
<span id="cb50-407"><a href="#cb50-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-408"><a href="#cb50-408" aria-hidden="true" tabindex="-1"></a>Order dependence with by="terms": If you care about marginal effects “as if added last,” refit with by="margin"(or reorder terms thoughtfully).</span>
<span id="cb50-409"><a href="#cb50-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-410"><a href="#cb50-410" aria-hidden="true" tabindex="-1"></a><span class="fu">## Fit betadispr</span></span>
<span id="cb50-411"><a href="#cb50-411" aria-hidden="true" tabindex="-1"></a><span class="in">`betadisper`</span> tests whether the within-group multivariate dispersion (spread) differs among groups. It’s the multivariate analogue of Levene’s test: it compares the distances of samples to their group centroid.</span>
<span id="cb50-412"><a href="#cb50-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-413"><a href="#cb50-413" aria-hidden="true" tabindex="-1"></a>What happens under the hood:</span>
<span id="cb50-414"><a href="#cb50-414" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Performs a PCoA on your distance matrix.</span>
<span id="cb50-415"><a href="#cb50-415" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Computes each group’s centroid in that ordination space.</span>
<span id="cb50-416"><a href="#cb50-416" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Calculates each sample’s distance to its group centroid.</span>
<span id="cb50-417"><a href="#cb50-417" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Stores those distances for downstream tests/plots.</span>
<span id="cb50-418"><a href="#cb50-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-419"><a href="#cb50-419" aria-hidden="true" tabindex="-1"></a>Define groups to test</span>
<span id="cb50-420"><a href="#cb50-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-423"><a href="#cb50-423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-424"><a href="#cb50-424" aria-hidden="true" tabindex="-1"></a>grp_leachate <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>leachate</span>
<span id="cb50-425"><a href="#cb50-425" aria-hidden="true" tabindex="-1"></a>grp_stage    <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>stage     </span>
<span id="cb50-426"><a href="#cb50-426" aria-hidden="true" tabindex="-1"></a>grp_int    <span class="ot">&lt;-</span> <span class="fu">interaction</span>(meta_filt<span class="sc">$</span>leachate, meta_filt<span class="sc">$</span>stage)</span>
<span id="cb50-427"><a href="#cb50-427" aria-hidden="true" tabindex="-1"></a>grp_night <span class="ot">&lt;-</span> meta_filt<span class="sc">$</span>spawn_night</span>
<span id="cb50-428"><a href="#cb50-428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-429"><a href="#cb50-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-430"><a href="#cb50-430" aria-hidden="true" tabindex="-1"></a><span class="fu">### night</span></span>
<span id="cb50-433"><a href="#cb50-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-434"><a href="#cb50-434" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit beta dispersion model </span></span>
<span id="cb50-435"><a href="#cb50-435" aria-hidden="true" tabindex="-1"></a>bd_night <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_night, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb50-436"><a href="#cb50-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-437"><a href="#cb50-437" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with no blocking</span></span>
<span id="cb50-438"><a href="#cb50-438" aria-hidden="true" tabindex="-1"></a>perm_night <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_night, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-439"><a href="#cb50-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-440"><a href="#cb50-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Result</span></span>
<span id="cb50-441"><a href="#cb50-441" aria-hidden="true" tabindex="-1"></a>perm_night</span>
<span id="cb50-442"><a href="#cb50-442" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-443"><a href="#cb50-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-444"><a href="#cb50-444" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb50-445"><a href="#cb50-445" aria-hidden="true" tabindex="-1"></a>multivariate dispersion differs across spawn nights (groups = nights). In other words, samples from some nights are more spread out around their night-specific centroid than others.</span>
<span id="cb50-446"><a href="#cb50-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-447"><a href="#cb50-447" aria-hidden="true" tabindex="-1"></a>betadisper with group = spawn_night significant heterogeneous dispersion among nights.If your primary tests are leachate/stage, treat night as a nuisance:</span>
<span id="cb50-448"><a href="#cb50-448" aria-hidden="true" tabindex="-1"></a>block by night in PERMANOVA (strata = spawn_night) so permutations happen within each night. This keeps night-to-night variance/dispersion from inflating your p-values for treatment effects.</span>
<span id="cb50-449"><a href="#cb50-449" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb50-450"><a href="#cb50-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-453"><a href="#cb50-453" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-454"><a href="#cb50-454" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_night, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb50-455"><a href="#cb50-455" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-456"><a href="#cb50-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-457"><a href="#cb50-457" aria-hidden="true" tabindex="-1"></a><span class="fu">### leachate</span></span>
<span id="cb50-460"><a href="#cb50-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-461"><a href="#cb50-461" aria-hidden="true" tabindex="-1"></a>bd_leachate <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_leachate, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb50-462"><a href="#cb50-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-463"><a href="#cb50-463" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb50-464"><a href="#cb50-464" aria-hidden="true" tabindex="-1"></a>perm_leachate <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_leachate, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>,</span>
<span id="cb50-465"><a href="#cb50-465" aria-hidden="true" tabindex="-1"></a>                  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night)  <span class="co"># optional</span></span>
<span id="cb50-466"><a href="#cb50-466" aria-hidden="true" tabindex="-1"></a>perm_leachate</span>
<span id="cb50-467"><a href="#cb50-467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-468"><a href="#cb50-468" aria-hidden="true" tabindex="-1"></a>What this tests:</span>
<span id="cb50-469"><a href="#cb50-469" aria-hidden="true" tabindex="-1"></a>Null hypothesis (Ho): all groups have equal average distance-to-centroid (equal dispersion/variance).If p &lt; 0.05, at least one group is more/less dispersed than another.Why use the permutation test? The distances depend on the PCoA, so permutation is safer than the plain ANOVA F-test.</span>
<span id="cb50-470"><a href="#cb50-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-473"><a href="#cb50-473" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-474"><a href="#cb50-474" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_leachate, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb50-475"><a href="#cb50-475" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-476"><a href="#cb50-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-477"><a href="#cb50-477" aria-hidden="true" tabindex="-1"></a><span class="fu">### stage</span></span>
<span id="cb50-480"><a href="#cb50-480" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-481"><a href="#cb50-481" aria-hidden="true" tabindex="-1"></a>bd_stage <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_stage, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb50-482"><a href="#cb50-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-483"><a href="#cb50-483" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb50-484"><a href="#cb50-484" aria-hidden="true" tabindex="-1"></a>perm_stage <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_stage, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-485"><a href="#cb50-485" aria-hidden="true" tabindex="-1"></a>                  <span class="co">#strata = meta_filt$spawn_night)  # optional</span></span>
<span id="cb50-486"><a href="#cb50-486" aria-hidden="true" tabindex="-1"></a>perm_stage</span>
<span id="cb50-487"><a href="#cb50-487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-488"><a href="#cb50-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-491"><a href="#cb50-491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-492"><a href="#cb50-492" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_stage, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb50-493"><a href="#cb50-493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-494"><a href="#cb50-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-495"><a href="#cb50-495" aria-hidden="true" tabindex="-1"></a><span class="fu">### leachate x stage</span></span>
<span id="cb50-498"><a href="#cb50-498" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-499"><a href="#cb50-499" aria-hidden="true" tabindex="-1"></a>bd_int <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitchison_dist, <span class="at">group =</span> grp_int, <span class="at">type =</span> <span class="st">"centroid"</span>)</span>
<span id="cb50-500"><a href="#cb50-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-501"><a href="#cb50-501" aria-hidden="true" tabindex="-1"></a><span class="co"># Permutation test with optional blocking</span></span>
<span id="cb50-502"><a href="#cb50-502" aria-hidden="true" tabindex="-1"></a>perm_int <span class="ot">&lt;-</span> <span class="fu">permutest</span>(bd_int, <span class="at">permutations =</span> <span class="dv">999</span>, <span class="at">pairwise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-503"><a href="#cb50-503" aria-hidden="true" tabindex="-1"></a>                  <span class="co">#strata = meta_filt$spawn_night)  # optional</span></span>
<span id="cb50-504"><a href="#cb50-504" aria-hidden="true" tabindex="-1"></a>perm_int</span>
<span id="cb50-505"><a href="#cb50-505" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-506"><a href="#cb50-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-509"><a href="#cb50-509" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-510"><a href="#cb50-510" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bd_int, <span class="at">hull =</span> <span class="cn">TRUE</span>, <span class="at">ellipse =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb50-511"><a href="#cb50-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-512"><a href="#cb50-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-513"><a href="#cb50-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-514"><a href="#cb50-514" aria-hidden="true" tabindex="-1"></a><span class="fu">## Perform PERMANOVA</span></span>
<span id="cb50-515"><a href="#cb50-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-516"><a href="#cb50-516" aria-hidden="true" tabindex="-1"></a>PERMANOVA mainly tests centroid location differences among groups (i.e., group separation).</span>
<span id="cb50-517"><a href="#cb50-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-518"><a href="#cb50-518" aria-hidden="true" tabindex="-1"></a>betadisper tests dispersion (variance) differences among groups.</span>
<span id="cb50-519"><a href="#cb50-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-520"><a href="#cb50-520" aria-hidden="true" tabindex="-1"></a>If PERMANOVA is significant and betadisper is non-significant, the signal likely reflects location (good).</span>
<span id="cb50-521"><a href="#cb50-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-522"><a href="#cb50-522" aria-hidden="true" tabindex="-1"></a>If both are significant, some of the PERMANOVA “difference” could be due to dispersion heterogeneity (groups with different spreads). Report both, and interpret cautiously.</span>
<span id="cb50-523"><a href="#cb50-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-526"><a href="#cb50-526" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-527"><a href="#cb50-527" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb50-528"><a href="#cb50-528" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PERMANOVA on Aitchison distance</span></span>
<span id="cb50-529"><a href="#cb50-529" aria-hidden="true" tabindex="-1"></a><span class="co"># Test effects of leachate, stage, and their interaction (blocking permutations by parental cross)</span></span>
<span id="cb50-530"><a href="#cb50-530" aria-hidden="true" tabindex="-1"></a>permanova_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(</span>
<span id="cb50-531"><a href="#cb50-531" aria-hidden="true" tabindex="-1"></a>  aitchison_dist <span class="sc">~</span> leachate <span class="sc">*</span> stage,</span>
<span id="cb50-532"><a href="#cb50-532" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> meta_filt,</span>
<span id="cb50-533"><a href="#cb50-533" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb50-534"><a href="#cb50-534" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"terms"</span>,</span>
<span id="cb50-535"><a href="#cb50-535" aria-hidden="true" tabindex="-1"></a>  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night</span>
<span id="cb50-536"><a href="#cb50-536" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-537"><a href="#cb50-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-538"><a href="#cb50-538" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(permanova_result)</span>
<span id="cb50-539"><a href="#cb50-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-540"><a href="#cb50-540" aria-hidden="true" tabindex="-1"></a><span class="co"># Save results</span></span>
<span id="cb50-541"><a href="#cb50-541" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(permanova_result), </span>
<span id="cb50-542"><a href="#cb50-542" aria-hidden="true" tabindex="-1"></a>          <span class="st">"../output/permanova/permanova_result_interactionblocked.csv"</span>,</span>
<span id="cb50-543"><a href="#cb50-543" aria-hidden="true" tabindex="-1"></a>          <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-544"><a href="#cb50-544" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-545"><a href="#cb50-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-548"><a href="#cb50-548" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-549"><a href="#cb50-549" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb50-550"><a href="#cb50-550" aria-hidden="true" tabindex="-1"></a><span class="co"># Additional PERMANOVA treating stage as continuous time </span></span>
<span id="cb50-551"><a href="#cb50-551" aria-hidden="true" tabindex="-1"></a>permanova_cont.time <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(</span>
<span id="cb50-552"><a href="#cb50-552" aria-hidden="true" tabindex="-1"></a>  aitchison_dist <span class="sc">~</span> leachate <span class="sc">*</span> hpf,</span>
<span id="cb50-553"><a href="#cb50-553" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> meta_filt,</span>
<span id="cb50-554"><a href="#cb50-554" aria-hidden="true" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb50-555"><a href="#cb50-555" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">"terms"</span>,</span>
<span id="cb50-556"><a href="#cb50-556" aria-hidden="true" tabindex="-1"></a>  <span class="at">strata =</span> meta_filt<span class="sc">$</span>spawn_night <span class="co"># control for parental cross</span></span>
<span id="cb50-557"><a href="#cb50-557" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-558"><a href="#cb50-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-559"><a href="#cb50-559" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(permanova_cont.time)</span>
<span id="cb50-560"><a href="#cb50-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-561"><a href="#cb50-561" aria-hidden="true" tabindex="-1"></a><span class="co"># Save results</span></span>
<span id="cb50-562"><a href="#cb50-562" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(permanova_cont.time), </span>
<span id="cb50-563"><a href="#cb50-563" aria-hidden="true" tabindex="-1"></a>          <span class="st">"../output/permanova/permanova_result_cont.time.csv"</span>,</span>
<span id="cb50-564"><a href="#cb50-564" aria-hidden="true" tabindex="-1"></a>          <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-565"><a href="#cb50-565" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-566"><a href="#cb50-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-567"><a href="#cb50-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-568"><a href="#cb50-568" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary</span></span>
<span id="cb50-569"><a href="#cb50-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-570"><a href="#cb50-570" aria-hidden="true" tabindex="-1"></a>This analysis implemented the key recommendations from Gloor et al. 2017 for compositional microbiome data analysis:</span>
<span id="cb50-571"><a href="#cb50-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-572"><a href="#cb50-572" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>**CLR Transformation**: Applied centered log-ratio transformation to convert compositional data to Euclidean space</span>
<span id="cb50-573"><a href="#cb50-573" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>**Aitchison Distance**: Calculated Euclidean distance on CLR-transformed data as the appropriate dissimilarity metric</span>
<span id="cb50-574"><a href="#cb50-574" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>**PCA Biplot**: Created variance-based PCA visualizations on CLR-transformed data</span>
<span id="cb50-575"><a href="#cb50-575" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>**PERMANOVA**: Tested for significant differences in microbial composition across experimental factors</span>
<span id="cb50-576"><a href="#cb50-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-577"><a href="#cb50-577" aria-hidden="true" tabindex="-1"></a><span class="fu">## Key Findings</span></span>
<span id="cb50-578"><a href="#cb50-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-579"><a href="#cb50-579" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PCA plots visually show clustering patterns by Developmental stage (aka timepoint/ hours post fertilization), Night of Spawn, and Parental cross (the later two are confounded)</span>
<span id="cb50-580"><a href="#cb50-580" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PERMANOVA results (blocked across Night of Spawn) indicate significant effects of Developmental stage (aka timepoint/ hours post fertilization), and non-significance for leachate and leahcate:stage interactions</span>
<span id="cb50-581"><a href="#cb50-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-582"><a href="#cb50-582" aria-hidden="true" tabindex="-1"></a><span class="fu">## Next Steps</span></span>
<span id="cb50-583"><a href="#cb50-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-584"><a href="#cb50-584" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Differential abundance analysis (MaAslin2)</span>
<span id="cb50-585"><a href="#cb50-585" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Conduct PERMANOVA on differentially abundant taxa</span>
<span id="cb50-586"><a href="#cb50-586" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Functional annotation of differentially abundant taxa (picrust)</span>
<span id="cb50-587"><a href="#cb50-587" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Differential abundance of functions (functional enrichment) (MaAslin2)</span>
<span id="cb50-588"><a href="#cb50-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-589"><a href="#cb50-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-590"><a href="#cb50-590" aria-hidden="true" tabindex="-1"></a><span class="fu">## Output Files</span></span>
<span id="cb50-591"><a href="#cb50-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-592"><a href="#cb50-592" aria-hidden="true" tabindex="-1"></a>All results and visualizations have been saved to the <span class="in">`../output/permanova`</span> directory:</span>
<span id="cb50-593"><a href="#cb50-593" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PCA biplots</span>
<span id="cb50-594"><a href="#cb50-594" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PERMANOVA results tables</span>
<span id="cb50-595"><a href="#cb50-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-596"><a href="#cb50-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-597"><a href="#cb50-597" aria-hidden="true" tabindex="-1"></a><span class="fu"># Session Info</span></span>
<span id="cb50-598"><a href="#cb50-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-601"><a href="#cb50-601" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-602"><a href="#cb50-602" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb50-603"><a href="#cb50-603" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




<script src="compositional_analysis_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>