<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sarah Tanja">
<meta name="dcterms.date" content="2026-02-23">

<title>Differential abundance of bacterial taxa</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="tpthalicius_files/libs/clipboard/clipboard.min.js"></script>
<script src="tpthalicius_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="tpthalicius_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="tpthalicius_files/libs/quarto-html/popper.min.js"></script>
<script src="tpthalicius_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="tpthalicius_files/libs/quarto-html/anchor.min.js"></script>
<link href="tpthalicius_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="tpthalicius_files/libs/quarto-html/quarto-syntax-highlighting-4a990d8dcb58f517c7c86712b8f2ac7c.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="tpthalicius_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="tpthalicius_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="tpthalicius_files/libs/bootstrap/bootstrap-9af6de2548c106c4a05dbc3b1b139c6f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background"><span class="header-section-number">1</span> Background</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup"><span class="header-section-number">2</span> Setup</a>
  <ul class="collapse">
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries"><span class="header-section-number">2.1</span> Load libraries</a></li>
  <li><a href="#define-file-paths" id="toc-define-file-paths" class="nav-link" data-scroll-target="#define-file-paths"><span class="header-section-number">2.2</span> Define file paths</a></li>
  <li><a href="#set-custom-ggplot-theme" id="toc-set-custom-ggplot-theme" class="nav-link" data-scroll-target="#set-custom-ggplot-theme"><span class="header-section-number">2.3</span> Set custom ggplot theme</a></li>
  <li><a href="#set-colorschemes" id="toc-set-colorschemes" class="nav-link" data-scroll-target="#set-colorschemes"><span class="header-section-number">2.4</span> Set colorschemes</a></li>
  </ul></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data"><span class="header-section-number">3</span> Load Data</a>
  <ul class="collapse">
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata"><span class="header-section-number">3.1</span> Metadata</a></li>
  <li><a href="#feature-table" id="toc-feature-table" class="nav-link" data-scroll-target="#feature-table"><span class="header-section-number">3.2</span> Feature table</a></li>
  <li><a href="#significant-results" id="toc-significant-results" class="nav-link" data-scroll-target="#significant-results"><span class="header-section-number">3.3</span> Significant results</a></li>
  </ul></li>
  <li><a href="#explore-significant-taxa" id="toc-explore-significant-taxa" class="nav-link" data-scroll-target="#explore-significant-taxa"><span class="header-section-number">4</span> Explore significant taxa</a>
  <ul class="collapse">
  <li><a href="#maaslin3-tropicobacter-phthalicius-results" id="toc-maaslin3-tropicobacter-phthalicius-results" class="nav-link" data-scroll-target="#maaslin3-tropicobacter-phthalicius-results"><span class="header-section-number">4.1</span> <code>MaAslin3</code> <em>Tropicobacter phthalicius</em> results</a></li>
  <li><a href="#plot-tropicobacter-phthalicius-across-development" id="toc-plot-tropicobacter-phthalicius-across-development" class="nav-link" data-scroll-target="#plot-tropicobacter-phthalicius-across-development"><span class="header-section-number">4.2</span> Plot <em>Tropicobacter phthalicius</em> across development</a></li>
  <li><a href="#relative-abundance-normalized-read-counts" id="toc-relative-abundance-normalized-read-counts" class="nav-link" data-scroll-target="#relative-abundance-normalized-read-counts"><span class="header-section-number">4.3</span> Relative abundance (normalized read counts)</a></li>
  <li><a href="#presenceabsence" id="toc-presenceabsence" class="nav-link" data-scroll-target="#presenceabsence"><span class="header-section-number">4.4</span> Presence/Absence</a></li>
  </ul></li>
  <li><a href="#explore-pattern-clustering" id="toc-explore-pattern-clustering" class="nav-link" data-scroll-target="#explore-pattern-clustering"><span class="header-section-number">5</span> Explore pattern clustering?</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="tpthalicius.md"><i class="bi bi-file-code"></i>Github (GFM)</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Differential abundance of bacterial taxa</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Exploring <em>Tropicobacter phthalicius</em> in the context of the microbiome</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Sarah Tanja </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 23, 2026</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">February 23, 2026</p>
    </div>
  </div>
    
  </div>
  


</header>


<section id="background" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Background</h1>
</section>
<section id="setup" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Setup</h1>
<section id="load-libraries" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="load-libraries"><span class="header-section-number">2.1</span> Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsidekick) <span class="co"># sean anderson ggplot themes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="define-file-paths" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="define-file-paths"><span class="header-section-number">2.2</span> Define file paths</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file paths</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>feature_table_path <span class="ot">&lt;-</span> <span class="st">"../../salipante/241121_StonyCoral/270x200/collapsed-l7.qza"</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>metadata_path <span class="ot">&lt;-</span> <span class="st">"../../metadata/meta.csv"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">"../../output/maaslin-taxa/"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>sig_results_path <span class="ot">&lt;-</span> <span class="st">"../../output/maaslin-taxa/Interaction/interaction_cat.con/significant_results.tsv"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-custom-ggplot-theme" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="set-custom-ggplot-theme"><span class="header-section-number">2.3</span> Set custom ggplot theme</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>theme_sleek_axe <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek</span>() <span class="sc">+</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"grey70"</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="cn">NA</span>,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">linewidth =</span> <span class="fl">0.5</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line.x  =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey70"</span>),</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line.y  =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey70"</span>),</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">size  =</span> <span class="dv">10</span>,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"grey40"</span>,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">b =</span> <span class="dv">20</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">size   =</span> <span class="dv">8</span>,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color  =</span> <span class="st">"grey40"</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">hjust  =</span> <span class="fl">0.5</span>,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">b =</span> <span class="dv">40</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">"grey40"</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-colorschemes" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="set-colorschemes"><span class="header-section-number">2.4</span> Set colorschemes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>leachate.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">control =</span> <span class="st">"#AEF1FF"</span>, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">low =</span> <span class="st">"#BBC7FF"</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">mid =</span> <span class="st">"#7D8BFF"</span>, </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">high =</span> <span class="st">"#592F7D"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>stage.<span class="fl">5.</span>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">egg =</span> <span class="st">"#FFE362"</span>,</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cleavage =</span> <span class="st">"#EBA600"</span>, </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">morula =</span> <span class="st">"#E6AA83"</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prawnchip =</span> <span class="st">"#D9685B"</span>, </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">earlygastrula =</span> <span class="st">"#A2223C"</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>stage.<span class="fl">3.</span>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">cleavage =</span> <span class="st">"#EBA600"</span>, </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">prawnchip =</span> <span class="st">"#D9685B"</span>, </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">earlygastrula =</span> <span class="st">"#A2223C"</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>status.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">typical =</span> <span class="st">"#75C165"</span>, </span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">uncertain =</span> <span class="st">"#E3FAA5"</span>, </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">malformed =</span> <span class="st">"#8B0069"</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>night.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">July_6th =</span> <span class="st">"#E3FAA5"</span>, </span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_7th =</span> <span class="st">"#578B21"</span>, </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_8th =</span> <span class="st">"#1E2440"</span>)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>night.colors.alt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">July_6th =</span> <span class="st">"#21918C"</span>, </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_7th =</span> <span class="st">"#201158"</span>, </span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_8th =</span> <span class="st">"#1E2440"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Load Data</h1>
<section id="metadata" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="metadata"><span class="header-section-number">3.1</span> Metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load metadata</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(metadata_path)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># set factors</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">collection_date =</span> <span class="fu">as.Date</span>(collection_date, <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">stage    =</span> <span class="fu">factor</span>(stage,    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"cleavage"</span>, <span class="st">"prawnchip"</span>, <span class="st">"earlygastrula"</span>), <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">leachate =</span> <span class="fu">factor</span>(leachate, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"control"</span>, <span class="st">"low"</span>, <span class="st">"mid"</span>, <span class="st">"high"</span>),        <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">spawn_night =</span> <span class="fu">factor</span>(</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>      collection_date,</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels  =</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"06-Jul-2024"</span>, <span class="st">"07-Jul-2024"</span>, <span class="st">"08-Jul-2024"</span>), <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels  =</span> <span class="fu">c</span>(<span class="st">"July 6th"</span>, <span class="st">"July 7th"</span>, <span class="st">"July 8th"</span>),</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">ordered =</span> <span class="cn">TRUE</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="feature-table" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="feature-table"><span class="header-section-number">3.2</span> Feature table</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load feature table from QIIME2 artifact</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>ft_l7 <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(feature_table_path)<span class="sc">$</span>data</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data.frame</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>ft_l7 <span class="ot">&lt;-</span> ft_l7 <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="significant-results" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="significant-results"><span class="header-section-number">3.3</span> Significant results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load significant results from MaAsLin3 output</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>sig_diff_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(sig_results_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="explore-significant-taxa" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Explore significant taxa</h1>
<p>Pull out significant taxa that are not flagged with an error (i.e., those that are truly significant and not due to some issue in the analysis). We will use this vector of significant taxa for further exploration.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>sig_diff_features_vector <span class="ot">&lt;-</span> sig_diff_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(error)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="fu">unique</span>(feature))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filter the feature table to subset only the significant taxa. In the feature table, the features are rownames.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sig_ft_l7 <span class="ot">&lt;-</span> ft_l7 <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"feature"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">%in%</span> sig_diff_features_vector)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(sig_ft_l7<span class="sc">$</span>feature))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 81</code></pre>
</div>
</div>
<p>Each row is a unique string.</p>
<p>For plotting we will need to pivot the data longer so that we have one row per sample and feature combination. This will allow us to easily plot the abundance of each significant taxon across our metadata variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>sig_ft_long <span class="ot">&lt;-</span> sig_ft_l7 <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>feature, <span class="co"># Keep 'feature' as is, pivot the rest</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"sample_id"</span>, <span class="co"># New column for sample IDs</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"rel_abundance"</span> <span class="co"># New column for relative (proportional) abundance values</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Combine the long feature table of significant taxa with the metadata. This will allow us to explore how the abundance of each significant taxon varies across our metadata variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>sig_ft_long_meta <span class="ot">&lt;-</span> sig_ft_long <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_id"</span> <span class="ot">=</span> <span class="st">"sample_id"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="maaslin3-tropicobacter-phthalicius-results" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="maaslin3-tropicobacter-phthalicius-results"><span class="header-section-number">4.1</span> <code>MaAslin3</code> <em>Tropicobacter phthalicius</em> results</h2>
<p>Let’s first pull out <em>Tropicobacter phthalicius</em> from the significant taxa and interpret the <code>MaAslin3</code> results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>sig_tropico <span class="ot">&lt;-</span> sig_diff_taxa <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">==</span> <span class="st">"d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus"</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>kableExtra<span class="sc">::</span><span class="fu">kable</span>(sig_tropico)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 47%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 3%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">feature</th>
<th style="text-align: left;">metadata</th>
<th style="text-align: left;">value</th>
<th style="text-align: left;">name</th>
<th style="text-align: right;">coef</th>
<th style="text-align: right;">null_hypothesis</th>
<th style="text-align: right;">stderr</th>
<th style="text-align: right;">pval_individual</th>
<th style="text-align: right;">qval_individual</th>
<th style="text-align: right;">pval_joint</th>
<th style="text-align: right;">qval_joint</th>
<th style="text-align: left;">error</th>
<th style="text-align: left;">model</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">N_not_zero</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.L</td>
<td style="text-align: left;">leachate.L</td>
<td style="text-align: right;">7.8738208</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1977069</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.L:hpf</td>
<td style="text-align: left;">leachate.L:hpf</td>
<td style="text-align: right;">7.2330406</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2655420</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: right;">5.5236586</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2636052</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.C</td>
<td style="text-align: left;">leachate.C</td>
<td style="text-align: right;">-4.6121759</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.3183642</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.C:hpf</td>
<td style="text-align: left;">leachate.C:hpf</td>
<td style="text-align: right;">-2.0345109</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2026284</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.Q</td>
<td style="text-align: left;">leachate.Q</td>
<td style="text-align: right;">-1.4373903</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2490276</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">8.00e-07</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000012</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">reads</td>
<td style="text-align: left;">reads</td>
<td style="text-align: left;">reads</td>
<td style="text-align: right;">0.9688052</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1986485</td>
<td style="text-align: right;">0.0000011</td>
<td style="text-align: right;">9.06e-05</td>
<td style="text-align: right;">2.2e-06</td>
<td style="text-align: right;">0.0001391</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">prevalence</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: left;">hpf</td>
<td style="text-align: right;">0.5738687</td>
<td style="text-align: right;">-0.2948786</td>
<td style="text-align: right;">0.1657204</td>
<td style="text-align: right;">0.0000193</td>
<td style="text-align: right;">1.45e-03</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.C</td>
<td style="text-align: left;">leachate.C</td>
<td style="text-align: right;">0.1841573</td>
<td style="text-align: right;">-0.1842725</td>
<td style="text-align: right;">0.2865674</td>
<td style="text-align: right;">0.2202621</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.C:hpf</td>
<td style="text-align: left;">leachate.C:hpf</td>
<td style="text-align: right;">-0.2375768</td>
<td style="text-align: right;">-0.1445292</td>
<td style="text-align: right;">0.2939018</td>
<td style="text-align: right;">0.7565845</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.L</td>
<td style="text-align: left;">leachate.L</td>
<td style="text-align: right;">-0.4510403</td>
<td style="text-align: right;">-0.0795706</td>
<td style="text-align: right;">0.2934988</td>
<td style="text-align: right;">0.2558601</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.L:hpf</td>
<td style="text-align: left;">leachate.L:hpf</td>
<td style="text-align: right;">-0.1795477</td>
<td style="text-align: right;">-0.0659666</td>
<td style="text-align: right;">0.3002812</td>
<td style="text-align: right;">0.7306502</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">leachate</td>
<td style="text-align: left;">.Q</td>
<td style="text-align: left;">leachate.Q</td>
<td style="text-align: right;">-0.1373198</td>
<td style="text-align: right;">-0.2115021</td>
<td style="text-align: right;">0.2902808</td>
<td style="text-align: right;">0.8121773</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">0.0e+00</td>
<td style="text-align: right;">0.0000012</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="even">
<td style="text-align: left;">d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</td>
<td style="text-align: left;">reads</td>
<td style="text-align: left;">reads</td>
<td style="text-align: left;">reads</td>
<td style="text-align: right;">-0.2659592</td>
<td style="text-align: right;">-0.0456286</td>
<td style="text-align: right;">0.1678833</td>
<td style="text-align: right;">0.2142442</td>
<td style="text-align: right;">1.00e+00</td>
<td style="text-align: right;">2.2e-06</td>
<td style="text-align: right;">0.0001391</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">abundance</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">52</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>We have N = 63 samples, and <em>T. phthalicicus</em> is present in 52 of them (N_not_zero = 52), so it’s fairly common.</li>
</ul>
<p>We’re looking at <code>MaAsLin3</code> hurdle model results … reference chatgpt interpretation <a href="https://chatgpt.com/share/699ca657-65c0-8008-80ce-083e7da498b4">here</a></p>
<section id="prevalence-model-logistic-model-presenceabsence" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="prevalence-model-logistic-model-presenceabsence"><span class="header-section-number">4.1.1</span> Prevalence model → logistic model (presence/absence)</h3>
<ul>
<li>This is a logistic regression, so coefficients are on the log-odds scale.</li>
</ul>
<p>##### leachate.L <strong>leachate.L | coef = 7.87, q = 0</strong> This is enormous and highly significant. A positive linear coefficient (7.87) means as leachate increases, the probability of detecting <em>Tropicibacter phthalicicus</em> increases strongly. 7.87 on the log-odds scale is huge.</p>
<p><span class="math display">\[
OR = e^{7.87} ≈ 2620
\]</span></p>
<p>That means:</p>
<p>Each step increase in leachate level massively increases odds of presence.</p>
<p>This is not subtle. This is explosive.</p>
<p>Because you used polynomial contrasts:</p>
<p>.L = linear trend across leachate levels</p>
<p>.Q = quadratic trend</p>
<p>.C = cubic trend</p>
<section id="leachate.lhpf" class="level5" data-number="4.1.1.0.1">
<h5 data-number="4.1.1.0.1" class="anchored" data-anchor-id="leachate.lhpf"><span class="header-section-number">4.1.1.0.1</span> leachate.L:hpf</h5>
<p><strong>leachate.L:hpf | coef = 7.23, q = 0</strong></p>
<p>This says:</p>
<p>The strength of the leachate effect depends on developmental stage.</p>
<p>Positive interaction means:</p>
<p>The leachate-driven increase in prevalence becomes stronger at later hpfs (with longer duration).</p>
</section>
<section id="hpf" class="level5" data-number="4.1.1.0.2">
<h5 data-number="4.1.1.0.2" class="anchored" data-anchor-id="hpf"><span class="header-section-number">4.1.1.0.2</span> hpf</h5>
<p><strong>hpf | coef = 5.52, q = 0</strong></p>
<p>Strong stage effect even independent of leachate. Developmental stage alone strongly affects whether this taxon is detected.</p>
<p>Quadratic and cubic leachate terms</p>
<p>.C = −4.61 (significant)</p>
<p>.Q = −1.44 (significant)</p>
<p>This tells you the response is not purely linear.</p>
<p>You likely have:</p>
<p>Strong increase at low → mid</p>
<p>Possibly leveling or non-monotonic behavior at higher levels</p>
<p>So prevalence probably curves rather than increases steadily. Higher sequencing depth increases probability of detecting it.</p>
<p>That’s expected.</p>
<p>But your leachate effect is far larger than read depth effect.</p>
<p>That’s reassuring.</p>
</section>
</section>
<section id="abundance-model-conditional-model-how-abundant-when-present" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="abundance-model-conditional-model-how-abundant-when-present"><span class="header-section-number">4.1.2</span> Abundance model → conditional model (how abundant when present)</h3>
<ul>
<li>This only models samples where taxon is nonzero.</li>
</ul>
</section>
<section id="big-biological-picture" class="level3" data-number="4.1.3">
<h3 data-number="4.1.3" class="anchored" data-anchor-id="big-biological-picture"><span class="header-section-number">4.1.3</span> Big Biological Picture</h3>
<p>This species is:</p>
<p>Highly responsive in presence/absence</p>
<p>Stage structured</p>
<p>But not meaningfully shifting in conditional abundance</p>
<p>That suggests:</p>
<p>Leachate exposure drives colonization or detection probability of Tropicibacter, rather than changing its growth intensity once established.</p>
<p>In coral embryo terms:</p>
<p>Possibly opportunistic colonizer under stress</p>
<p>Or suppressed in controls and released under disturbance</p>
<p>Strong stage modulation suggests developmental window sensitivity This is a classic hurdle model pattern:</p>
<p>Strong signal in prevalence</p>
<p>Weak/no signal in abundance</p>
<p>That means your taxon behaves like a binary ecological responder, not a graded responder.</p>
<p>It either shows up or it doesn’t.</p>
<p><img src="%22../../output/maaslin-taxa/Interaction/interaction_cat.con/figures/summary_plot.png%22" class="img-fluid"></p>
<p>If you’re looking at the MaAsLin3 heatmap:</p>
<p>Prevalence panel = log-odds coefficients</p>
<p>Abundance panel = conditional abundance coefficients</p>
<p>Large red squares in prevalence for leachate = strong increased detection odds Small or pale abundance squares = little effect on intensity</p>
<p>So your heatmap is visually encoding:</p>
<p>Leachate drives presence, not abundance magnitude.</p>
<section id="hpf-1" class="level5" data-number="4.1.3.0.1">
<h5 data-number="4.1.3.0.1" class="anchored" data-anchor-id="hpf-1"><span class="header-section-number">4.1.3.0.1</span> hpf</h5>
<p><strong>hpf | coef = 0.57, q = 0.001</strong> When present, abundance increases modestly with developmental stage.</p>
</section>
</section>
</section>
<section id="plot-tropicobacter-phthalicius-across-development" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="plot-tropicobacter-phthalicius-across-development"><span class="header-section-number">4.2</span> Plot <em>Tropicobacter phthalicius</em> across development</h2>
<p>The character string is: <code>d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for Tropicobacter phthalicius</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>tropico_ft <span class="ot">&lt;-</span> sig_ft_long_meta <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">==</span> <span class="st">"d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="relative-abundance-normalized-read-counts" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="relative-abundance-normalized-read-counts"><span class="header-section-number">4.3</span> Relative abundance (normalized read counts)</h2>
<p>Plot the relative abundance of <em>Tropicobacter phthalicius</em> across the different stages of development. We will use a boxplot to visualize the distribution of relative abundance for each stage.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(hpf), <span class="at">y =</span> rel_abundance, <span class="at">color =</span> leachate, <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span> <span class="co"># Add jitter for individual points</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Relative Abundance of Tropicobacter phthalicius in 14 hours of embryonic development"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Hours Post Fertilization"</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Relative Abundance (normalized read counts)"</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tpthalicius_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(hpf), <span class="at">y =</span> rel_abundance, <span class="at">color =</span> leachate, <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_jitter(alpha = 0.9, position = position_dodge(width = 0.8)) + # Add jitter for individual points</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Relative Abundance of Tropicobacter phthalicius in 14 hours of embryonic development"</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Hours Post Fertilization"</span>,</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Relative Abundance (normalized read counts)"</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tpthalicius_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="presenceabsence" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="presenceabsence"><span class="header-section-number">4.4</span> Presence/Absence</h2>
<p>Make a new column for presence/absence of <em>Tropicobacter phthalicius</em> (1 if rel_abundance &gt; 0, else 0). Then plot the prevalence of <em>Tropicobacter phthalicius</em> across developmental stages and leachate levels.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>tropico_ft <span class="ot">&lt;-</span> tropico_ft <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">presence =</span> <span class="fu">ifelse</span>(rel_abundance <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> leachate,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">as.numeric</span>(presence),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> leachate,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>),</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"glm"</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family =</span> <span class="st">"binomial"</span>),</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"grey70"</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"grey90"</span>,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="fl">0.8</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_dotplot</span>(</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">binaxis =</span> <span class="st">"y"</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">stackdir =</span> <span class="st">"centerwhole"</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.6</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">dotsize =</span> <span class="fl">0.6</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mean points</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> mean,</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"point"</span>,</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="dv">8</span>,</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>hpf) <span class="sc">+</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Prevalence of Tropicobacter phthalicius across leachate levels and developmental stages"</span>,</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Leachate Level"</span>,</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Prevalence (proportion of samples with presence)"</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="tpthalicius_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="explore-pattern-clustering" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Explore pattern clustering?</h1>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb19" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Differential abundance of bacterial taxa"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Exploring *Tropicobacter phthalicius* in the context of the microbiome"</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Sarah Tanja"</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2026-02-23</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="an">date-modified:</span><span class="co"> today</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">  gfm: </span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: journal</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: article</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-background: true</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: </span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co">      source: true</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">      toggle: true</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 2</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: left</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: kable</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co">    smooth-scroll: true</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co">    link-external-icon: true</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="co">    link-external-newwindow: true</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co">    reference-location: margin</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a><span class="co">    citation-location: margin</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a><span class="co">  markdown: </span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="co">    wrap: 72</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="co"># Display code chunks</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>, <span class="co"># Evaluate code chunks</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>, <span class="co"># Hide warnings</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>, <span class="co"># Hide messages</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">comment =</span> <span class="st">""</span> <span class="co"># Prevents appending '##' to beginning of lines in code output</span></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>)        </span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a><span class="fu"># Background</span></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a><span class="fu"># Setup</span></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="fu">## Load libraries</span></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsidekick) <span class="co"># sean anderson ggplot themes</span></span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a><span class="fu">## Define file paths</span></span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file paths</span></span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a>feature_table_path <span class="ot">&lt;-</span> <span class="st">"../../salipante/241121_StonyCoral/270x200/collapsed-l7.qza"</span></span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a>metadata_path <span class="ot">&lt;-</span> <span class="st">"../../metadata/meta.csv"</span></span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>output_path <span class="ot">&lt;-</span> <span class="st">"../../output/maaslin-taxa/"</span></span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>sig_results_path <span class="ot">&lt;-</span> <span class="st">"../../output/maaslin-taxa/Interaction/interaction_cat.con/significant_results.tsv"</span></span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a><span class="fu">## Set custom ggplot theme</span></span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a>theme_sleek_axe <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek</span>() <span class="sc">+</span></span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(</span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"grey70"</span>,</span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="cn">NA</span>,</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a>        <span class="at">linewidth =</span> <span class="fl">0.5</span></span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line.x  =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey70"</span>),</span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.line.y  =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey70"</span>),</span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(</span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">size  =</span> <span class="dv">10</span>,</span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"grey40"</span>,</span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a>        <span class="at">hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a>        <span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">b =</span> <span class="dv">20</span>)</span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(</span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>        <span class="at">size   =</span> <span class="dv">8</span>,</span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a>        <span class="at">color  =</span> <span class="st">"grey40"</span>,</span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>        <span class="at">hjust  =</span> <span class="fl">0.5</span>,</span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a>        <span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">b =</span> <span class="dv">40</span>)</span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb19-98"><a href="#cb19-98" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb19-99"><a href="#cb19-99" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb19-100"><a href="#cb19-100" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb19-101"><a href="#cb19-101" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">color =</span> <span class="st">"grey40"</span>),</span>
<span id="cb19-102"><a href="#cb19-102" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>, <span class="at">color =</span> <span class="st">"grey40"</span>)</span>
<span id="cb19-103"><a href="#cb19-103" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-104"><a href="#cb19-104" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-105"><a href="#cb19-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-106"><a href="#cb19-106" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-107"><a href="#cb19-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-108"><a href="#cb19-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-109"><a href="#cb19-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-110"><a href="#cb19-110" aria-hidden="true" tabindex="-1"></a><span class="fu">## Set colorschemes</span></span>
<span id="cb19-113"><a href="#cb19-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-114"><a href="#cb19-114" aria-hidden="true" tabindex="-1"></a>leachate.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">control =</span> <span class="st">"#AEF1FF"</span>, </span>
<span id="cb19-115"><a href="#cb19-115" aria-hidden="true" tabindex="-1"></a>                <span class="at">low =</span> <span class="st">"#BBC7FF"</span>,</span>
<span id="cb19-116"><a href="#cb19-116" aria-hidden="true" tabindex="-1"></a>                <span class="at">mid =</span> <span class="st">"#7D8BFF"</span>, </span>
<span id="cb19-117"><a href="#cb19-117" aria-hidden="true" tabindex="-1"></a>                <span class="at">high =</span> <span class="st">"#592F7D"</span>)</span>
<span id="cb19-118"><a href="#cb19-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-119"><a href="#cb19-119" aria-hidden="true" tabindex="-1"></a>stage.<span class="fl">5.</span>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">egg =</span> <span class="st">"#FFE362"</span>,</span>
<span id="cb19-120"><a href="#cb19-120" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cleavage =</span> <span class="st">"#EBA600"</span>, </span>
<span id="cb19-121"><a href="#cb19-121" aria-hidden="true" tabindex="-1"></a>                    <span class="at">morula =</span> <span class="st">"#E6AA83"</span>,</span>
<span id="cb19-122"><a href="#cb19-122" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prawnchip =</span> <span class="st">"#D9685B"</span>, </span>
<span id="cb19-123"><a href="#cb19-123" aria-hidden="true" tabindex="-1"></a>                    <span class="at">earlygastrula =</span> <span class="st">"#A2223C"</span>)</span>
<span id="cb19-124"><a href="#cb19-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-125"><a href="#cb19-125" aria-hidden="true" tabindex="-1"></a>stage.<span class="fl">3.</span>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">cleavage =</span> <span class="st">"#EBA600"</span>, </span>
<span id="cb19-126"><a href="#cb19-126" aria-hidden="true" tabindex="-1"></a>                  <span class="at">prawnchip =</span> <span class="st">"#D9685B"</span>, </span>
<span id="cb19-127"><a href="#cb19-127" aria-hidden="true" tabindex="-1"></a>                  <span class="at">earlygastrula =</span> <span class="st">"#A2223C"</span>)</span>
<span id="cb19-128"><a href="#cb19-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-129"><a href="#cb19-129" aria-hidden="true" tabindex="-1"></a>status.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">typical =</span> <span class="st">"#75C165"</span>, </span>
<span id="cb19-130"><a href="#cb19-130" aria-hidden="true" tabindex="-1"></a>                   <span class="at">uncertain =</span> <span class="st">"#E3FAA5"</span>, </span>
<span id="cb19-131"><a href="#cb19-131" aria-hidden="true" tabindex="-1"></a>                   <span class="at">malformed =</span> <span class="st">"#8B0069"</span>)</span>
<span id="cb19-132"><a href="#cb19-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-133"><a href="#cb19-133" aria-hidden="true" tabindex="-1"></a>night.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">July_6th =</span> <span class="st">"#E3FAA5"</span>, </span>
<span id="cb19-134"><a href="#cb19-134" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_7th =</span> <span class="st">"#578B21"</span>, </span>
<span id="cb19-135"><a href="#cb19-135" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_8th =</span> <span class="st">"#1E2440"</span>)</span>
<span id="cb19-136"><a href="#cb19-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-137"><a href="#cb19-137" aria-hidden="true" tabindex="-1"></a>night.colors.alt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">July_6th =</span> <span class="st">"#21918C"</span>, </span>
<span id="cb19-138"><a href="#cb19-138" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_7th =</span> <span class="st">"#201158"</span>, </span>
<span id="cb19-139"><a href="#cb19-139" aria-hidden="true" tabindex="-1"></a>                  <span class="at">July_8th =</span> <span class="st">"#1E2440"</span>)</span>
<span id="cb19-140"><a href="#cb19-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-141"><a href="#cb19-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-142"><a href="#cb19-142" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load Data</span></span>
<span id="cb19-143"><a href="#cb19-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-144"><a href="#cb19-144" aria-hidden="true" tabindex="-1"></a><span class="fu">## Metadata</span></span>
<span id="cb19-145"><a href="#cb19-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-148"><a href="#cb19-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-149"><a href="#cb19-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Load metadata</span></span>
<span id="cb19-150"><a href="#cb19-150" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(metadata_path)</span>
<span id="cb19-151"><a href="#cb19-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-152"><a href="#cb19-152" aria-hidden="true" tabindex="-1"></a><span class="co"># set factors</span></span>
<span id="cb19-153"><a href="#cb19-153" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb19-154"><a href="#cb19-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-155"><a href="#cb19-155" aria-hidden="true" tabindex="-1"></a>    <span class="at">collection_date =</span> <span class="fu">as.Date</span>(collection_date, <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb19-156"><a href="#cb19-156" aria-hidden="true" tabindex="-1"></a>    <span class="at">stage    =</span> <span class="fu">factor</span>(stage,    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"cleavage"</span>, <span class="st">"prawnchip"</span>, <span class="st">"earlygastrula"</span>), <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb19-157"><a href="#cb19-157" aria-hidden="true" tabindex="-1"></a>    <span class="at">leachate =</span> <span class="fu">factor</span>(leachate, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"control"</span>, <span class="st">"low"</span>, <span class="st">"mid"</span>, <span class="st">"high"</span>),        <span class="at">ordered =</span> <span class="cn">TRUE</span>),</span>
<span id="cb19-158"><a href="#cb19-158" aria-hidden="true" tabindex="-1"></a>    <span class="at">spawn_night =</span> <span class="fu">factor</span>(</span>
<span id="cb19-159"><a href="#cb19-159" aria-hidden="true" tabindex="-1"></a>      collection_date,</span>
<span id="cb19-160"><a href="#cb19-160" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels  =</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"06-Jul-2024"</span>, <span class="st">"07-Jul-2024"</span>, <span class="st">"08-Jul-2024"</span>), <span class="at">format =</span> <span class="st">"%d-%b-%Y"</span>),</span>
<span id="cb19-161"><a href="#cb19-161" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels  =</span> <span class="fu">c</span>(<span class="st">"July 6th"</span>, <span class="st">"July 7th"</span>, <span class="st">"July 8th"</span>),</span>
<span id="cb19-162"><a href="#cb19-162" aria-hidden="true" tabindex="-1"></a>      <span class="at">ordered =</span> <span class="cn">TRUE</span></span>
<span id="cb19-163"><a href="#cb19-163" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-164"><a href="#cb19-164" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb19-165"><a href="#cb19-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-166"><a href="#cb19-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-167"><a href="#cb19-167" aria-hidden="true" tabindex="-1"></a><span class="fu">## Feature table</span></span>
<span id="cb19-168"><a href="#cb19-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-171"><a href="#cb19-171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-172"><a href="#cb19-172" aria-hidden="true" tabindex="-1"></a><span class="co"># Load feature table from QIIME2 artifact</span></span>
<span id="cb19-173"><a href="#cb19-173" aria-hidden="true" tabindex="-1"></a>ft_l7 <span class="ot">&lt;-</span> <span class="fu">read_qza</span>(feature_table_path)<span class="sc">$</span>data</span>
<span id="cb19-174"><a href="#cb19-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-175"><a href="#cb19-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data.frame</span></span>
<span id="cb19-176"><a href="#cb19-176" aria-hidden="true" tabindex="-1"></a>ft_l7 <span class="ot">&lt;-</span> ft_l7 <span class="sc">%&gt;%</span> </span>
<span id="cb19-177"><a href="#cb19-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb19-178"><a href="#cb19-178" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-179"><a href="#cb19-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-180"><a href="#cb19-180" aria-hidden="true" tabindex="-1"></a><span class="fu">## Significant results </span></span>
<span id="cb19-183"><a href="#cb19-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-184"><a href="#cb19-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Load significant results from MaAsLin3 output</span></span>
<span id="cb19-185"><a href="#cb19-185" aria-hidden="true" tabindex="-1"></a>sig_diff_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(sig_results_path)</span>
<span id="cb19-186"><a href="#cb19-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-187"><a href="#cb19-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-188"><a href="#cb19-188" aria-hidden="true" tabindex="-1"></a><span class="fu"># Explore significant taxa</span></span>
<span id="cb19-189"><a href="#cb19-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-190"><a href="#cb19-190" aria-hidden="true" tabindex="-1"></a>Pull out significant taxa that are not flagged with an error (i.e., those that are truly significant and not due to some issue in the analysis). We will use this vector of significant taxa for further exploration.</span>
<span id="cb19-193"><a href="#cb19-193" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-194"><a href="#cb19-194" aria-hidden="true" tabindex="-1"></a>sig_diff_features_vector <span class="ot">&lt;-</span> sig_diff_taxa <span class="sc">%&gt;%</span> </span>
<span id="cb19-195"><a href="#cb19-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(error)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-196"><a href="#cb19-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="fu">unique</span>(feature))</span>
<span id="cb19-197"><a href="#cb19-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-198"><a href="#cb19-198" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-199"><a href="#cb19-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-200"><a href="#cb19-200" aria-hidden="true" tabindex="-1"></a>Filter the feature table to subset only the significant taxa. In the feature table, the features are rownames. </span>
<span id="cb19-201"><a href="#cb19-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-204"><a href="#cb19-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-205"><a href="#cb19-205" aria-hidden="true" tabindex="-1"></a>sig_ft_l7 <span class="ot">&lt;-</span> ft_l7 <span class="sc">%&gt;%</span> </span>
<span id="cb19-206"><a href="#cb19-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"feature"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-207"><a href="#cb19-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">%in%</span> sig_diff_features_vector)</span>
<span id="cb19-208"><a href="#cb19-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-209"><a href="#cb19-209" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(sig_ft_l7<span class="sc">$</span>feature))</span>
<span id="cb19-210"><a href="#cb19-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-211"><a href="#cb19-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-212"><a href="#cb19-212" aria-hidden="true" tabindex="-1"></a>Each row is a unique string. </span>
<span id="cb19-213"><a href="#cb19-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-214"><a href="#cb19-214" aria-hidden="true" tabindex="-1"></a>For plotting we will need to pivot the data longer so that we have one row per sample and feature combination. This will allow us to easily plot the abundance of each significant taxon across our metadata variables.</span>
<span id="cb19-215"><a href="#cb19-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-218"><a href="#cb19-218" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-219"><a href="#cb19-219" aria-hidden="true" tabindex="-1"></a>sig_ft_long <span class="ot">&lt;-</span> sig_ft_l7 <span class="sc">%&gt;%</span></span>
<span id="cb19-220"><a href="#cb19-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb19-221"><a href="#cb19-221" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">-</span>feature, <span class="co"># Keep 'feature' as is, pivot the rest</span></span>
<span id="cb19-222"><a href="#cb19-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"sample_id"</span>, <span class="co"># New column for sample IDs</span></span>
<span id="cb19-223"><a href="#cb19-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"rel_abundance"</span> <span class="co"># New column for relative (proportional) abundance values</span></span>
<span id="cb19-224"><a href="#cb19-224" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb19-225"><a href="#cb19-225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-226"><a href="#cb19-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-227"><a href="#cb19-227" aria-hidden="true" tabindex="-1"></a>Combine the long feature table of significant taxa with the metadata. This will allow us to explore how the abundance of each significant taxon varies across our metadata variables.</span>
<span id="cb19-230"><a href="#cb19-230" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-231"><a href="#cb19-231" aria-hidden="true" tabindex="-1"></a>sig_ft_long_meta <span class="ot">&lt;-</span> sig_ft_long <span class="sc">%&gt;%</span></span>
<span id="cb19-232"><a href="#cb19-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(meta, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_id"</span> <span class="ot">=</span> <span class="st">"sample_id"</span>))</span>
<span id="cb19-233"><a href="#cb19-233" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-234"><a href="#cb19-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-235"><a href="#cb19-235" aria-hidden="true" tabindex="-1"></a><span class="fu">## `MaAslin3` *Tropicobacter phthalicius* results</span></span>
<span id="cb19-236"><a href="#cb19-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-237"><a href="#cb19-237" aria-hidden="true" tabindex="-1"></a>Let's first pull out *Tropicobacter phthalicius* from the significant taxa and interpret the <span class="in">`MaAslin3`</span> results.</span>
<span id="cb19-238"><a href="#cb19-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-241"><a href="#cb19-241" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-242"><a href="#cb19-242" aria-hidden="true" tabindex="-1"></a>sig_tropico <span class="ot">&lt;-</span> sig_diff_taxa <span class="sc">%&gt;%</span></span>
<span id="cb19-243"><a href="#cb19-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">==</span> <span class="st">"d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus"</span>)</span>
<span id="cb19-244"><a href="#cb19-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-245"><a href="#cb19-245" aria-hidden="true" tabindex="-1"></a>kableExtra<span class="sc">::</span><span class="fu">kable</span>(sig_tropico)</span>
<span id="cb19-246"><a href="#cb19-246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-247"><a href="#cb19-247" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>We have N = 63 samples, and *T. phthalicicus* is present in 52 of them (N_not_zero = 52), so it’s fairly common.</span>
<span id="cb19-248"><a href="#cb19-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-249"><a href="#cb19-249" aria-hidden="true" tabindex="-1"></a>We're looking at <span class="in">`MaAsLin3`</span> hurdle model results ... reference chatgpt interpretation <span class="co">[</span><span class="ot">here</span><span class="co">](https://chatgpt.com/share/699ca657-65c0-8008-80ce-083e7da498b4)</span></span>
<span id="cb19-250"><a href="#cb19-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-251"><a href="#cb19-251" aria-hidden="true" tabindex="-1"></a><span class="fu">### Prevalence model → logistic model (presence/absence)</span></span>
<span id="cb19-252"><a href="#cb19-252" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>This is a logistic regression, so coefficients are on the log-odds scale.</span>
<span id="cb19-253"><a href="#cb19-253" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-254"><a href="#cb19-254" aria-hidden="true" tabindex="-1"></a>  ##### leachate.L </span>
<span id="cb19-255"><a href="#cb19-255" aria-hidden="true" tabindex="-1"></a>**leachate.L | coef = 7.87, q = 0**</span>
<span id="cb19-256"><a href="#cb19-256" aria-hidden="true" tabindex="-1"></a>This is enormous and highly significant. A positive linear coefficient (7.87) means as leachate increases, the probability of detecting *Tropicibacter phthalicicus* increases strongly. 7.87 on the log-odds scale is huge.</span>
<span id="cb19-257"><a href="#cb19-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-258"><a href="#cb19-258" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb19-259"><a href="#cb19-259" aria-hidden="true" tabindex="-1"></a>OR = e^{7.87} ≈ 2620</span>
<span id="cb19-260"><a href="#cb19-260" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb19-261"><a href="#cb19-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-262"><a href="#cb19-262" aria-hidden="true" tabindex="-1"></a>That means:</span>
<span id="cb19-263"><a href="#cb19-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-264"><a href="#cb19-264" aria-hidden="true" tabindex="-1"></a>Each step increase in leachate level massively increases odds of presence.</span>
<span id="cb19-265"><a href="#cb19-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-266"><a href="#cb19-266" aria-hidden="true" tabindex="-1"></a>This is not subtle. This is explosive.</span>
<span id="cb19-267"><a href="#cb19-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-268"><a href="#cb19-268" aria-hidden="true" tabindex="-1"></a>Because you used polynomial contrasts:</span>
<span id="cb19-269"><a href="#cb19-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-270"><a href="#cb19-270" aria-hidden="true" tabindex="-1"></a>.L = linear trend across leachate levels</span>
<span id="cb19-271"><a href="#cb19-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-272"><a href="#cb19-272" aria-hidden="true" tabindex="-1"></a>.Q = quadratic trend</span>
<span id="cb19-273"><a href="#cb19-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-274"><a href="#cb19-274" aria-hidden="true" tabindex="-1"></a>.C = cubic trend</span>
<span id="cb19-275"><a href="#cb19-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-276"><a href="#cb19-276" aria-hidden="true" tabindex="-1"></a><span class="fu">##### leachate.L:hpf</span></span>
<span id="cb19-277"><a href="#cb19-277" aria-hidden="true" tabindex="-1"></a>**leachate.L:hpf | coef = 7.23, q = 0**</span>
<span id="cb19-278"><a href="#cb19-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-279"><a href="#cb19-279" aria-hidden="true" tabindex="-1"></a>This says:</span>
<span id="cb19-280"><a href="#cb19-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-281"><a href="#cb19-281" aria-hidden="true" tabindex="-1"></a>The strength of the leachate effect depends on developmental stage.</span>
<span id="cb19-282"><a href="#cb19-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-283"><a href="#cb19-283" aria-hidden="true" tabindex="-1"></a>Positive interaction means:</span>
<span id="cb19-284"><a href="#cb19-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-285"><a href="#cb19-285" aria-hidden="true" tabindex="-1"></a>The leachate-driven increase in prevalence becomes stronger at later hpfs (with longer duration). </span>
<span id="cb19-286"><a href="#cb19-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-287"><a href="#cb19-287" aria-hidden="true" tabindex="-1"></a><span class="fu">##### hpf</span></span>
<span id="cb19-288"><a href="#cb19-288" aria-hidden="true" tabindex="-1"></a>**hpf | coef = 5.52, q = 0**</span>
<span id="cb19-289"><a href="#cb19-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-290"><a href="#cb19-290" aria-hidden="true" tabindex="-1"></a>Strong stage effect even independent of leachate. Developmental stage alone strongly affects whether this taxon is detected.</span>
<span id="cb19-291"><a href="#cb19-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-292"><a href="#cb19-292" aria-hidden="true" tabindex="-1"></a>Quadratic and cubic leachate terms</span>
<span id="cb19-293"><a href="#cb19-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-294"><a href="#cb19-294" aria-hidden="true" tabindex="-1"></a>.C = −4.61 (significant)</span>
<span id="cb19-295"><a href="#cb19-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-296"><a href="#cb19-296" aria-hidden="true" tabindex="-1"></a>.Q = −1.44 (significant)</span>
<span id="cb19-297"><a href="#cb19-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-298"><a href="#cb19-298" aria-hidden="true" tabindex="-1"></a>This tells you the response is not purely linear.</span>
<span id="cb19-299"><a href="#cb19-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-300"><a href="#cb19-300" aria-hidden="true" tabindex="-1"></a>You likely have:</span>
<span id="cb19-301"><a href="#cb19-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-302"><a href="#cb19-302" aria-hidden="true" tabindex="-1"></a>Strong increase at low → mid</span>
<span id="cb19-303"><a href="#cb19-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-304"><a href="#cb19-304" aria-hidden="true" tabindex="-1"></a>Possibly leveling or non-monotonic behavior at higher levels</span>
<span id="cb19-305"><a href="#cb19-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-306"><a href="#cb19-306" aria-hidden="true" tabindex="-1"></a>So prevalence probably curves rather than increases steadily.</span>
<span id="cb19-307"><a href="#cb19-307" aria-hidden="true" tabindex="-1"></a>Higher sequencing depth increases probability of detecting it.</span>
<span id="cb19-308"><a href="#cb19-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-309"><a href="#cb19-309" aria-hidden="true" tabindex="-1"></a>That’s expected.</span>
<span id="cb19-310"><a href="#cb19-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-311"><a href="#cb19-311" aria-hidden="true" tabindex="-1"></a>But your leachate effect is far larger than read depth effect.</span>
<span id="cb19-312"><a href="#cb19-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-313"><a href="#cb19-313" aria-hidden="true" tabindex="-1"></a>That’s reassuring.</span>
<span id="cb19-314"><a href="#cb19-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-315"><a href="#cb19-315" aria-hidden="true" tabindex="-1"></a><span class="fu">### Abundance model → conditional model (how abundant when present)</span></span>
<span id="cb19-316"><a href="#cb19-316" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>This only models samples where taxon is nonzero.</span>
<span id="cb19-317"><a href="#cb19-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-318"><a href="#cb19-318" aria-hidden="true" tabindex="-1"></a><span class="fu">### Big Biological Picture</span></span>
<span id="cb19-319"><a href="#cb19-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-320"><a href="#cb19-320" aria-hidden="true" tabindex="-1"></a>This species is:</span>
<span id="cb19-321"><a href="#cb19-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-322"><a href="#cb19-322" aria-hidden="true" tabindex="-1"></a>Highly responsive in presence/absence</span>
<span id="cb19-323"><a href="#cb19-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-324"><a href="#cb19-324" aria-hidden="true" tabindex="-1"></a>Stage structured</span>
<span id="cb19-325"><a href="#cb19-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-326"><a href="#cb19-326" aria-hidden="true" tabindex="-1"></a>But not meaningfully shifting in conditional abundance</span>
<span id="cb19-327"><a href="#cb19-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-328"><a href="#cb19-328" aria-hidden="true" tabindex="-1"></a>That suggests:</span>
<span id="cb19-329"><a href="#cb19-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-330"><a href="#cb19-330" aria-hidden="true" tabindex="-1"></a>Leachate exposure drives colonization or detection probability of Tropicibacter, rather than changing its growth intensity once established.</span>
<span id="cb19-331"><a href="#cb19-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-332"><a href="#cb19-332" aria-hidden="true" tabindex="-1"></a>In coral embryo terms:</span>
<span id="cb19-333"><a href="#cb19-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-334"><a href="#cb19-334" aria-hidden="true" tabindex="-1"></a>Possibly opportunistic colonizer under stress</span>
<span id="cb19-335"><a href="#cb19-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-336"><a href="#cb19-336" aria-hidden="true" tabindex="-1"></a>Or suppressed in controls and released under disturbance</span>
<span id="cb19-337"><a href="#cb19-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-338"><a href="#cb19-338" aria-hidden="true" tabindex="-1"></a>Strong stage modulation suggests developmental window sensitivity</span>
<span id="cb19-339"><a href="#cb19-339" aria-hidden="true" tabindex="-1"></a>This is a classic hurdle model pattern:</span>
<span id="cb19-340"><a href="#cb19-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-341"><a href="#cb19-341" aria-hidden="true" tabindex="-1"></a>Strong signal in prevalence</span>
<span id="cb19-342"><a href="#cb19-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-343"><a href="#cb19-343" aria-hidden="true" tabindex="-1"></a>Weak/no signal in abundance</span>
<span id="cb19-344"><a href="#cb19-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-345"><a href="#cb19-345" aria-hidden="true" tabindex="-1"></a>That means your taxon behaves like a binary ecological responder, not a graded responder.</span>
<span id="cb19-346"><a href="#cb19-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-347"><a href="#cb19-347" aria-hidden="true" tabindex="-1"></a>It either shows up or it doesn’t.</span>
<span id="cb19-348"><a href="#cb19-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-349"><a href="#cb19-349" aria-hidden="true" tabindex="-1"></a><span class="al">![]("../../output/maaslin-taxa/Interaction/interaction_cat.con/figures/summary_plot.png")</span></span>
<span id="cb19-350"><a href="#cb19-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-351"><a href="#cb19-351" aria-hidden="true" tabindex="-1"></a>If you’re looking at the MaAsLin3 heatmap:</span>
<span id="cb19-352"><a href="#cb19-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-353"><a href="#cb19-353" aria-hidden="true" tabindex="-1"></a>Prevalence panel = log-odds coefficients</span>
<span id="cb19-354"><a href="#cb19-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-355"><a href="#cb19-355" aria-hidden="true" tabindex="-1"></a>Abundance panel = conditional abundance coefficients</span>
<span id="cb19-356"><a href="#cb19-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-357"><a href="#cb19-357" aria-hidden="true" tabindex="-1"></a>Large red squares in prevalence for leachate = strong increased detection odds</span>
<span id="cb19-358"><a href="#cb19-358" aria-hidden="true" tabindex="-1"></a>Small or pale abundance squares = little effect on intensity</span>
<span id="cb19-359"><a href="#cb19-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-360"><a href="#cb19-360" aria-hidden="true" tabindex="-1"></a>So your heatmap is visually encoding:</span>
<span id="cb19-361"><a href="#cb19-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-362"><a href="#cb19-362" aria-hidden="true" tabindex="-1"></a>Leachate drives presence, not abundance magnitude.</span>
<span id="cb19-363"><a href="#cb19-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-364"><a href="#cb19-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-365"><a href="#cb19-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-366"><a href="#cb19-366" aria-hidden="true" tabindex="-1"></a><span class="fu">##### hpf</span></span>
<span id="cb19-367"><a href="#cb19-367" aria-hidden="true" tabindex="-1"></a>**hpf | coef = 0.57, q = 0.001**</span>
<span id="cb19-368"><a href="#cb19-368" aria-hidden="true" tabindex="-1"></a>When present, abundance increases modestly with developmental stage.</span>
<span id="cb19-369"><a href="#cb19-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-370"><a href="#cb19-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-371"><a href="#cb19-371" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plot *Tropicobacter phthalicius* across development</span></span>
<span id="cb19-372"><a href="#cb19-372" aria-hidden="true" tabindex="-1"></a>The character string is:</span>
<span id="cb19-373"><a href="#cb19-373" aria-hidden="true" tabindex="-1"></a><span class="in">`d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus`</span></span>
<span id="cb19-374"><a href="#cb19-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-377"><a href="#cb19-377" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-378"><a href="#cb19-378" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for Tropicobacter phthalicius</span></span>
<span id="cb19-379"><a href="#cb19-379" aria-hidden="true" tabindex="-1"></a>tropico_ft <span class="ot">&lt;-</span> sig_ft_long_meta <span class="sc">%&gt;%</span></span>
<span id="cb19-380"><a href="#cb19-380" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(feature <span class="sc">==</span> <span class="st">"d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Tropicibacter;s__Tropicibacter_phthalicicus"</span>)</span>
<span id="cb19-381"><a href="#cb19-381" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-382"><a href="#cb19-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-383"><a href="#cb19-383" aria-hidden="true" tabindex="-1"></a><span class="fu">## Relative abundance (normalized read counts)</span></span>
<span id="cb19-384"><a href="#cb19-384" aria-hidden="true" tabindex="-1"></a>Plot the relative abundance of *Tropicobacter phthalicius* across the different stages of development. We will use a boxplot to visualize the distribution of relative abundance for each stage.</span>
<span id="cb19-387"><a href="#cb19-387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-388"><a href="#cb19-388" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(hpf), <span class="at">y =</span> rel_abundance, <span class="at">color =</span> leachate, <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb19-389"><a href="#cb19-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb19-390"><a href="#cb19-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span> <span class="co"># Add jitter for individual points</span></span>
<span id="cb19-391"><a href="#cb19-391" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-392"><a href="#cb19-392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-393"><a href="#cb19-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-394"><a href="#cb19-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Relative Abundance of Tropicobacter phthalicius in 14 hours of embryonic development"</span>,</span>
<span id="cb19-395"><a href="#cb19-395" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Hours Post Fertilization"</span>,</span>
<span id="cb19-396"><a href="#cb19-396" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Relative Abundance (normalized read counts)"</span></span>
<span id="cb19-397"><a href="#cb19-397" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-398"><a href="#cb19-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span>
<span id="cb19-399"><a href="#cb19-399" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-402"><a href="#cb19-402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-403"><a href="#cb19-403" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(hpf), <span class="at">y =</span> rel_abundance, <span class="at">color =</span> leachate, <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb19-404"><a href="#cb19-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.8</span>)) <span class="sc">+</span></span>
<span id="cb19-405"><a href="#cb19-405" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_jitter(alpha = 0.9, position = position_dodge(width = 0.8)) + # Add jitter for individual points</span></span>
<span id="cb19-406"><a href="#cb19-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-407"><a href="#cb19-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-408"><a href="#cb19-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-409"><a href="#cb19-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Relative Abundance of Tropicobacter phthalicius in 14 hours of embryonic development"</span>,</span>
<span id="cb19-410"><a href="#cb19-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Hours Post Fertilization"</span>,</span>
<span id="cb19-411"><a href="#cb19-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Relative Abundance (normalized read counts)"</span></span>
<span id="cb19-412"><a href="#cb19-412" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-413"><a href="#cb19-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span>
<span id="cb19-414"><a href="#cb19-414" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-415"><a href="#cb19-415" aria-hidden="true" tabindex="-1"></a><span class="fu">## Presence/Absence</span></span>
<span id="cb19-416"><a href="#cb19-416" aria-hidden="true" tabindex="-1"></a>Make a new column for presence/absence of *Tropicobacter phthalicius* (1 if rel_abundance &gt; 0, else 0). Then plot the prevalence of *Tropicobacter phthalicius* across developmental stages and leachate levels.</span>
<span id="cb19-417"><a href="#cb19-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-420"><a href="#cb19-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-421"><a href="#cb19-421" aria-hidden="true" tabindex="-1"></a>tropico_ft <span class="ot">&lt;-</span> tropico_ft <span class="sc">%&gt;%</span></span>
<span id="cb19-422"><a href="#cb19-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">presence =</span> <span class="fu">ifelse</span>(rel_abundance <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb19-423"><a href="#cb19-423" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-424"><a href="#cb19-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-427"><a href="#cb19-427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-428"><a href="#cb19-428" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tropico_ft,</span>
<span id="cb19-429"><a href="#cb19-429" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> leachate,</span>
<span id="cb19-430"><a href="#cb19-430" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">as.numeric</span>(presence),</span>
<span id="cb19-431"><a href="#cb19-431" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> leachate,</span>
<span id="cb19-432"><a href="#cb19-432" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill =</span> leachate)) <span class="sc">+</span></span>
<span id="cb19-433"><a href="#cb19-433" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-434"><a href="#cb19-434" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(</span>
<span id="cb19-435"><a href="#cb19-435" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>),</span>
<span id="cb19-436"><a href="#cb19-436" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"glm"</span>,</span>
<span id="cb19-437"><a href="#cb19-437" aria-hidden="true" tabindex="-1"></a>    <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family =</span> <span class="st">"binomial"</span>),</span>
<span id="cb19-438"><a href="#cb19-438" aria-hidden="true" tabindex="-1"></a>    <span class="at">se =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-439"><a href="#cb19-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"grey70"</span>,</span>
<span id="cb19-440"><a href="#cb19-440" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"grey90"</span>,</span>
<span id="cb19-441"><a href="#cb19-441" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="fl">0.8</span></span>
<span id="cb19-442"><a href="#cb19-442" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-443"><a href="#cb19-443" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-444"><a href="#cb19-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_dotplot</span>(</span>
<span id="cb19-445"><a href="#cb19-445" aria-hidden="true" tabindex="-1"></a>    <span class="at">binaxis =</span> <span class="st">"y"</span>,</span>
<span id="cb19-446"><a href="#cb19-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">stackdir =</span> <span class="st">"centerwhole"</span>,</span>
<span id="cb19-447"><a href="#cb19-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.6</span>,</span>
<span id="cb19-448"><a href="#cb19-448" aria-hidden="true" tabindex="-1"></a>    <span class="at">dotsize =</span> <span class="fl">0.6</span></span>
<span id="cb19-449"><a href="#cb19-449" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-450"><a href="#cb19-450" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mean points</span></span>
<span id="cb19-451"><a href="#cb19-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb19-452"><a href="#cb19-452" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> mean,</span>
<span id="cb19-453"><a href="#cb19-453" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"point"</span>,</span>
<span id="cb19-454"><a href="#cb19-454" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="dv">8</span>,</span>
<span id="cb19-455"><a href="#cb19-455" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span></span>
<span id="cb19-456"><a href="#cb19-456" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-457"><a href="#cb19-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-458"><a href="#cb19-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>hpf) <span class="sc">+</span></span>
<span id="cb19-459"><a href="#cb19-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-460"><a href="#cb19-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> leachate.colors) <span class="sc">+</span></span>
<span id="cb19-461"><a href="#cb19-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-462"><a href="#cb19-462" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Prevalence of Tropicobacter phthalicius across leachate levels and developmental stages"</span>,</span>
<span id="cb19-463"><a href="#cb19-463" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Leachate Level"</span>,</span>
<span id="cb19-464"><a href="#cb19-464" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Prevalence (proportion of samples with presence)"</span></span>
<span id="cb19-465"><a href="#cb19-465" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-466"><a href="#cb19-466" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sleek_axe</span>()</span>
<span id="cb19-467"><a href="#cb19-467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-468"><a href="#cb19-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-469"><a href="#cb19-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-470"><a href="#cb19-470" aria-hidden="true" tabindex="-1"></a><span class="fu"># Explore pattern clustering?</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




<script src="tpthalicius_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>